---
title: "CASE Full FINAL"
output: github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message = TRUE)
```
# Setup
```{r include=TRUE}
library(ggplot2)
library(tidyr)
library(plyr)
library(dplyr)
library(qvalue)
library(stringr)
library(pcadapt)
library(poolSeq)
library(ACER)
library(ggman)
library(VennDiagram)
library(scales)
library(data.table)
library(patchwork)


# Custom theme for black background figures
theme_black = function(base_size = 16, base_family = "") {
  
  theme_grey(base_size = base_size, base_family = base_family) %+replace%
    
    theme(
      # Specify axis options
      axis.line = element_line(colour = "white", size = 1.5),
      axis.text.x = element_text(size = base_size*0.8, color = "white", lineheight = 0.9),  
      axis.text.y = element_text(size = base_size*0.8, color = "white", lineheight = 0.9),  
      axis.ticks = element_line(color = "white", size  =  0.2),  
      axis.title.x = element_text(size = base_size, color = "white", margin = margin(0, 10, 0, 0)),  
      axis.title.y = element_text(size = base_size, color = "white", angle = 90, margin = margin(0, 10, 0, 0)),  
      axis.ticks.length = unit(0.3, "lines"),   
      # Specify legend options
      legend.background = element_rect(color = NA, fill = "transparent"),  
      legend.key = element_rect(color = NA,  fill = "transparent"),  
      legend.key.size = unit(1.2, "lines"),  
      legend.key.height = NULL,  
      legend.key.width = NULL,      
      legend.text = element_text(size = base_size*0.8, color = "white"),  
      legend.title = element_text(size = base_size*0.8, face = "bold", hjust = 0, color = "white"),  
      legend.position = "right",  
      legend.text.align = NULL,  
      legend.title.align = NULL,  
      legend.direction = "vertical",  
      legend.box = NULL, 
      # Specify panel options
      panel.background = element_rect(fill = "transparent", color  =  NA),  
      panel.border = element_blank(),  
      panel.grid.major = element_line(color = "transparent"),  
      panel.grid.minor = element_line(color = "transparent"),  
      #panel.margin = unit(0.5, "lines"),   
      panel.spacing= unit(0.5, "lines"),
      # Specify facetting options
      strip.background = element_rect(fill = "transparent", color = "transparent"),  
      strip.text.x = element_text(size = base_size*0.8, color = "white"),  
      strip.text.y = element_text(size = base_size*0.8, color = "white",angle = -90),  
      # Specify plot options
      plot.background = element_rect(color = "transparent", fill = "transparent"),  
      plot.title = element_text(size = base_size*1.2, color = "white", hjust = 1),  
      plot.margin = unit(rep(1, 4), "lines")
      
    )
  
}

# Function to convert p-values to q-values
Qvalue_convert <- function(table) {


table <- spread(table,GROUP,PVAL)  
  
table$CHR <- table$CHROM

table %>% 
  mutate(CHR = str_replace(CHR, "NC_035780.1", "1")) %>% 
  mutate(CHR = str_replace(CHR, "NC_035781.1", "2")) %>% 
  mutate(CHR = str_replace(CHR, "NC_035782.1", "3")) %>% 
  mutate(CHR = str_replace(CHR, "NC_035783.1", "4")) %>% 
  mutate(CHR = str_replace(CHR, "NC_035784.1", "5")) %>% 
  mutate(CHR = str_replace(CHR, "NC_035785.1", "6")) %>% 
  mutate(CHR = str_replace(CHR, "NC_035786.1", "7")) %>% 
  mutate(CHR = str_replace(CHR, "NC_035787.1", "8")) %>%
  mutate(CHR = str_replace(CHR, "NC_035788.1", "9")) %>% 
  mutate(CHR = str_replace(CHR, "NC_035789.1", "10"))  -> pp1

pp1 <- unite(pp1, "SNP", c("CHROM","BP"),remove = FALSE)

pp1$CHR <- as.numeric(pp1$CHR)  

# Set pi0=1 to equal Benjamini Hochberg
pp1$QCA <- qvalue(pp1$PCA, pi0 = 1)$qvalues
pp1$QCON <- qvalue(pp1$PCON, pi0 = 1)$qvalues
pp1$QSE <- qvalue(pp1$PSE, pi0 = 1)$qvalues
pp1$QCASE <- qvalue(pp1$PCASE, pi0 = 1)$qvalues

#pp1$QCA <- qvalue(pp1$PCA)$qvalues
#pp1$QCON <- qvalue(pp1$PCON)$qvalues
#pp1$QSE <- qvalue(pp1$PSE)$qvalues
#pp1$QCASE <- qvalue(pp1$PCASE)$qvalues

return(pp1)
}

# Function to determine significant loci in various outpus
Significat_subset <- function(pv, alpha, alpha2) {
  # Add new logical columns based on conditions
  pv$Sig.CASE <- pv$QCASE < alpha
  pv$Sig.CA <- pv$QCA < alpha
  pv$Sig.SE <- pv$QSE < alpha
  
  # Subset the dataframe
  ppsig <- subset(pv, QCA < alpha | QCASE < alpha | QSE < alpha)
  ppsig <- subset(ppsig, QCON > alpha2)
  
  # Print diagnostics
  print(nrow(pv))
  print(nrow(ppsig))
  print(nrow(ppsig) / nrow(pv))
  
  # Return the modified dataframe
  return(ppsig)
}

group_and_average <- function(df) {
  df %>%
    dplyr::group_by(SNP) %>%
    dplyr::summarize(across(c(PCA, PCASE, PCON, PSE, CHR, QCA, QCON, QSE, QCASE), 
                            ~ if (all(is.na(.))) NA else min(., na.rm = TRUE)),
                     Sig.CASE = any(Sig.CASE, na.rm = TRUE),
                     Sig.CA = any(Sig.CA, na.rm = TRUE),Sig.SE = any(Sig.SE, na.rm = TRUE),
                     CHROM = first(CHROM), BP = first(BP), .groups = "drop") }


cbPaletteSmall <- c("#E69F00", "#009E73", "#999999","sky blue","#0072B2")
cbPaletteSmall4 <- c("#999999","#E69F00", "#0072B2", "#009E73")
cbPaletteSmall3 <- c("#E69F00", "#0072B2","#009E73")
```

```{r}
sessionInfo()
```

## Download Popoolation2 scripts
```{bash, eval = FALSE}
cd ../scripts
git clone https://github.com/ToBoDev/assessPool.git
```

## Create conda (mamba) environment
```{bash, eval=FALSE}
mamba env create --file ../CASE_environment.yaml
mamba env create --file ../random_draw_environment.yaml
```
# Filtering
```{bash, eval=FALSE}
source activate CASE
cd ../raw.vcf
bash ../scripts/CASE_PVCF_filter2.sh CASE 64 20
```


```{bash, eval=FALSE}
source activate CASE
cd ../raw.vcf 
bcftools index CASE.TRSdp.20.g5.nDNA.FIL.vcf.gz
bcftools query -l CASE.TRSdp.20.g5.nDNA.FIL.vcf.gz > samples
```
## Create Unified Sync file
Each sequencing run had some extra samples that need to be removed before starting the analysis
Each block was filtered for missing data less than 10% and MAF of > 0.015 based on read counts using VCF file first.



```{bash, eval=FALSE}
source activate CASE

cd ../raw.vcf
bcftools view -S <(grep B11 samples | grep -v 1.G) CASE.TRSdp.20.g5.nDNA.FIL.vcf.gz | bcftools view -i 'F_MISSING<0.1'  | bcftools view -M 4 -m 2 | bcftools +fill-tags -- -t 'AAF:1=sum(FORMAT/AO)/sum(FORMAT/DP)' | bcftools +fill-tags -- -t  FORMAT/VAF | bcftools view --threads 40 -i 'AAF > 0.015 && AAF < 0.985' | bcftools query -f '%CHROM\t%POS\n' > B11.pos

bcftools view -S <(grep B10 samples | grep -v J17 | grep -v J05) CASE.TRSdp.20.g5.nDNA.FIL.vcf.gz | bcftools view -i 'F_MISSING<0.1' | bcftools view -M 4 -m 2 | bcftools +fill-tags -- -t 'AAF:1=sum(FORMAT/AO)/sum(FORMAT/DP)' | bcftools +fill-tags -- -t  FORMAT/VAF | bcftools view --threads 40 -i 'AAF > 0.015 && AAF < 0.985' | bcftools query -f '%CHROM\t%POS\n' > B10.pos

bcftools view -S <(grep B12 samples) CASE.TRSdp.20.g5.nDNA.FIL.vcf.gz | bcftools view -i 'F_MISSING<0.1' | bcftools view -M 4 -m 2 | bcftools +fill-tags -- -t 'AAF:1=sum(FORMAT/AO)/sum(FORMAT/DP)' | bcftools +fill-tags -- -t  FORMAT/VAF | bcftools view --threads 40 -i 'AAF > 0.015 && AAF < 0.985' | bcftools query -f '%CHROM\t%POS\n'  > B12.pos

cat B*.pos | sort | uniq > fil.pos

bcftools view -R fil.pos -S <(grep -v 1.GB11 samples | grep -v J17B10| grep -v J05B10) --threads 40 -m2 -M 4 CASE.TRSdp.20.g5.nDNA.FIL.vcf.gz -O z -o SNP.CASE.TRSdp.20.B90.2a.perp.vcf.gz

#bcftools view --threads 40 SNP.CASE.TRSdp.20.B90.2a.perp.vcf.gz | mawk '!/#/' | cut -f 1,2 | mawk '{print $1"\t"$2-1"\t"$2}' > total.snp.bed

#bedtools intersect -wb -a total.snp.bed -b ~/CASE/analysis/sorted.ref3.0.gene.bed | grep -oh "gene=LOC.*;g" | sed 's/gene=//g' | sed 's/;g//g' | sort | uniq > CASE.study.background.LOC

```

### Create Sync files
```{bash, eval=FALSE}
source activate CASE

bcftools view --threads 40 ../raw.vcf/SNP.CASE.TRSdp.20.B90.2a.perp.vcf.gz  | mawk -F'\t' -v OFS='\t' '{ for(i=1;i<=NF;i++) if($i==".:.:.:.:.:.:.:.") $i="."; print }' > temp.vcf
python2 ../scripts/VCFtoPopPool.py temp.vcf CASE.All.Blocks.sync 
rm temp.vcf
```

### This sort is to maintain reproducibility across older analysis
```{bash}
mawk 'BEGIN {OFS="\t"
    order["CHROM"] = 0
    order["NC_035789.1"] = 1
    order["NC_035780.1"] = 2
    order["NC_035781.1"] = 3
    order["NC_035782.1"] = 4
    order["NC_035783.1"] = 5
    order["NC_035784.1"] = 6
    order["NC_035785.1"] = 7
    order["NC_035786.1"] = 8
    order["NC_035787.1"] = 9
    order["NC_035788.1"] = 10
}
{ print order[$1], $2, NR, $0 }' CASE.All.Blocks.sync | sort -k1,1n -k2,2n -k3,3n | cut -f4- > test.sync

mv test.sync CASE.All.Blocks.sync
```


```{bash}
cat <(echo -e "CHROM\tPOS") ../raw.vcf/B10.pos > B10.pos
cat <(echo -e "CHROM\tPOS") ../raw.vcf/B11.pos > B11.pos
cat <(echo -e "CHROM\tPOS") ../raw.vcf/B12.pos > B12.pos

mawk 'NR==FNR{a[$1,$2]; next} ($1,$2) in a' B10.pos CASE.All.Blocks.sync | mawk 'NR==1{for(i=1;i<=NF;i++)if(i<=3||$i~/B10$/)k[i]}{o="";for(i=1;i<=NF;i++)if(i in k)o=o?o OFS $i:$i;print o}' OFS='\t' | mawk '!/\t\.\t/' > CASE.Block10.sync
mawk 'NR==FNR{a[$1,$2]; next} ($1,$2) in a' B11.pos CASE.All.Blocks.sync | mawk 'NR==1{for(i=1;i<=NF;i++)if(i<=3||$i~/B11$/)k[i]}{o="";for(i=1;i<=NF;i++)if(i in k)o=o?o OFS $i:$i;print o}' OFS='\t' | mawk '!/\t\.\t/' > CASE.Block11.sync
mawk 'NR==FNR{a[$1,$2]; next} ($1,$2) in a' B12.pos CASE.All.Blocks.sync | mawk 'NR==1{for(i=1;i<=NF;i++)if(i<=3||$i~/B12$/)k[i]}{o="";for(i=1;i<=NF;i++)if(i in k)o=o?o OFS $i:$i;print o}' OFS='\t' | mawk '!/\t\.\t/' > CASE.Block12.sync
```

## Add coverage stats to sync file and filter by minimum coverage

```{bash, eval=FALSE}
source activate CASE
mawk -f ../scripts/add_cov_sync CASE.Block10.sync | mawk '$20 > 9 && $22 > 24' > CASE.dp20.Block10.cov.sync &
mawk -f ../scripts/add_cov_sync CASE.Block11.sync | mawk '$24 > 9 && $26 > 24' > CASE.dp20.Block11.cov.sync &
mawk -f ../scripts/add_cov_sync CASE.Block12.sync | mawk '$24 > 9 && $26 > 24' > CASE.dp20.Block12.cov.sync &
#mawk -f ../scripts/add_cov_sync CASE.All.Blocks.sync | mawk '$66 > 9 && $68 > 24' > CASE.All.Blocks.cov.sync
mawk -f ../scripts/add_cov_sync CASE.All.Blocks.sync | mawk '$60 > 9 && $62 > 24' | mawk '!/\t\.\t/'  > CASE.All.Blocks.cov.sync

cut -f1,2 CASE.dp20.Block1*.cov.sync | sort | uniq -c | mawk '$1 > 2' | mawk '{print $2 "\t" $3}' > filtered.loci.allthreeblocks
```

# Visualize across 10,000 random loci

## All
```{bash, eval = FALSE}
source activate CASE

mawk '!/CHR/' filtered.loci.allthreeblocks | shuf -n 50000  | shuf -n 10000 > rand.loci

mawk 'NR==FNR{a[$1,$2]; next} ($1,$2) in a' rand.loci CASE.All.Blocks.cov.sync | cut --complement -f60- > input.all.rand.sync

```

```{r}
input.all.rand <- read.sync(file="input.all.rand.sync", gen=seq(1,56), repl=seq(1,56), polarization="rising")


input.all.rand.af <- af(input.all.rand,gen=seq(1,56), repl=seq(1,56))
treatment.labels <- c(rep("CA", 11),rep("CASE", 11),rep("CON", 11),rep("IS", 12),rep("SE", 11))
spawn.names <- c("B12","B11","B12","B10","B11","B10","B11","B12","B10","B12","B11","B10","B11","B12","B12","B10","B12","B11","B10","B11","B11","B12","B10","B11","B12","B11","B10","B12","B10","B11","B12","B11","B12","B10","B11","B10","B11","B10","B11","B10","B11","B12","B12","B12","B12","B11","B12","B10","B10","B11","B11","B10","B12","B12","B11","B12")

colnames(input.all.rand.af) <- paste(treatment.labels,spawn.names,sep="_")


# Perform PCA
af_t <- t(input.all.rand.af)

pca_res <- prcomp(af_t, center = TRUE, scale. = TRUE)

# Get scores for plotting
pca_df <- as.data.frame(pca_res$x)
#pca_df$locus <- rownames(input.all.rand.af)  # keep locus IDs if you have them

var_explained <- pca_res$sdev^2 / sum(pca_res$sdev^2) * 100

# Example: if you have a vector "pop_group" the same length as rows

treatment.labels <- c(rep("CA", 11),rep("CASE", 11),rep("CON", 11),rep("IS", 12),rep("SE", 11))
spawn.names <- c("B12","B11","B12","B10","B11","B10","B11","B12","B10","B12","B11","B10","B11","B12","B12","B10","B12","B11","B10","B11","B11","B12","B10","B11","B12","B11","B10","B12","B10","B11","B12","B11","B12","B10","B11","B10","B11","B10","B11","B10","B11","B12","B12","B12","B12","B11","B12","B10","B10","B11","B11","B10","B12","B12","B11","B12")

pca_df$Population <- spawn.names
pca_df$Treatments <- treatment.labels

ggplot(pca_df, aes(x = PC1, y = PC2, color = Population, fill=Population)) +
  geom_point(aes(alpha=0.2), size =10,shape=21, col="black")+  scale_fill_manual(values=cbPaletteSmall,name="Treatment")+ 
guides(alpha=FALSE) +theme(axis.title.x = element_text(size = 24),axis.title.y = element_text(size = 24)) +
  theme_minimal() +
  labs(
    x = paste0("PC1 (", round(var_explained[1], 1), "%)"),
    y = paste0("PC2 (", round(var_explained[2], 1), "%)"),
    title = "PCA of Allele Frequencies"
  )

```
```{r}
png(filename="PC_all_rand1.png", type="cairo",units="px", width=5400, height=3000, res=300, bg="transparent")
ggplot(pca_df, aes(x=PC1, y= PC2, color = Treatments, fill=Treatments)) + geom_point(aes(alpha=0.2), size =10,shape=21, col="black")+ theme_classic() + scale_fill_manual(values=cbPaletteSmall,name="Treatment")+ 
guides(alpha=FALSE) +theme(axis.title.x = element_text(size = 24),axis.title.y = element_text(size = 24))+
  labs(
    x = paste0("PC1 (", round(var_explained[1], 1), "%)"),
    y = paste0("PC2 (", round(var_explained[2], 1), "%)"),
    title = "PCA of Allele Frequencies"
  )
dev.off()
png(filename="PC_all_rand1S.png", type="cairo",units="px", width=5400, height=3000, res=300, bg="transparent")
ggplot(pca_df, aes(x=PC1, y= PC2, color = Population, fill=Population)) + geom_point(aes(alpha=0.2), size =10,shape=21, col="black")+ theme_classic() + scale_fill_manual(values=cbPaletteSmall,name="Spawn")+ 
guides(alpha=FALSE) +theme(axis.title.x = element_text(size = 24),axis.title.y = element_text(size = 24))+
  labs(
    x = paste0("PC1 (", round(var_explained[1], 1), "%)"),
    y = paste0("PC2 (", round(var_explained[2], 1), "%)"),
    title = "PCA of Allele Frequencies"
  )
dev.off()
```

## All Initial Samples

```{r}
input.IS.rand <- input.all.rand.af[,34:45]
afi_t <- t(input.IS.rand)

pca_i_res <- prcomp(afi_t, center = TRUE, scale. = TRUE)

# Get scores for plotting
pca_i_df <- as.data.frame(pca_i_res$x)

var_explained <- pca_i_res$sdev^2 / sum(pca_i_res$sdev^2) * 100

pca_i_df$Population <- spawn.names[34:45]
pca_i_df$Treatments <- c(rep("IS", 12))

ggplot(pca_i_df, aes(x = PC1, y = PC2, color = Population, fill=Population)) +
  geom_point(aes(alpha=0.2), size =10,shape=21, col="black")+  scale_fill_manual(values=cbPaletteSmall,name="Treatment")+ 
guides(alpha=FALSE) +theme(axis.title.x = element_text(size = 24),axis.title.y = element_text(size = 24)) +
  theme_classic() +
  labs(
    x = paste0("PC1 (", round(var_explained[1], 1), "%)"),
    y = paste0("PC2 (", round(var_explained[2], 1), "%)"),
    title = "PCA of Allele Frequencies"
  )

```



```{r}
png(filename="PC_all_IS_rand1.png", type="cairo",units="px", width=5400, height=3000, res=300, bg="transparent")

ggplot(pca_i_df, aes(x = PC1, y = PC2, color = Population, fill=Population)) +
  geom_point(aes(alpha=0.2), size =10,shape=21, col="black")+  scale_fill_manual(values=cbPaletteSmall,name="Spawn")+ 
guides(alpha=FALSE) +theme(axis.title.x = element_text(size = 24),axis.title.y = element_text(size = 24)) +
  theme_classic() +
  labs(
    x = paste0("PC1 (", round(var_explained[1], 1), "%)"),
    y = paste0("PC2 (", round(var_explained[2], 1), "%)"),
    title = "PCA of Allele Frequencies"
  )
```

## Random loci for each individual blocks
```{r}

# Optional: clean whitespace from column names (safe-guard)
colnames(input.all.rand.af) <- trimws(colnames(input.all.rand.af))

stages <- c("B10","B11","B12")

for(stage in stages){
  # match columns that end with "_B10" / "_B11" / "_B12"
  cols_idx <- grep(paste0("_", stage, "$"), colnames(input.all.rand.af))
  cat(sprintf("\n== Stage %s: found %d sample columns ==\n", stage, length(cols_idx)))
  if(length(cols_idx) < 2){
    warning(sprintf("Stage %s: less than 2 samples found - skipping PCA.", stage)); next
  }

  sub_mat <- input.all.rand.af[, cols_idx, drop = FALSE]
  cat("  dim(sub_mat) (loci x samples):", dim(sub_mat), "\n")

  # transpose so rows = samples, cols = loci (variables)
  mat_t <- t(sub_mat)

  # remove loci (columns) with zero variance across the samples in this stage
  var_cols <- apply(mat_t, 2, var, na.rm = TRUE)
  keep <- which(var_cols > 0)
  removed_n <- sum(var_cols == 0)
  cat("  loci removed (zero variance):", removed_n, "\n")

  if(length(keep) < 2){
    warning(sprintf("Stage %s: fewer than 2 variable loci after filtering - skipping PCA.", stage)); next
  }
  mat_t2 <- mat_t[, keep, drop = FALSE]
  cat("  dim(mat_t2) (samples x retained loci):", dim(mat_t2), "\n")

  # run PCA (now safe to scale)
  pca_res <- prcomp(mat_t2, center = TRUE, scale. = TRUE)

  # PCA dataframe
  pca_df <- as.data.frame(pca_res$x)

  # extract Treatments and Spawn from sample names (rownames of mat_t2)
  sample_names <- rownames(mat_t2)
  pca_df$Treatment <- sub("_.*", "", sample_names)   # text before first "_"
  pca_df$Spawn     <- sub(".*_", "", sample_names)   # text after last "_"

  # variance explained
  var_explained <- (pca_res$sdev^2) / sum(pca_res$sdev^2) * 100

  # choose palette fallback if cbPaletteSmall not defined
  if(exists("cbPaletteSmall")){
    fill_scale <- scale_fill_manual(values = cbPaletteSmall, name = "Treatment")
  } else {
    fill_scale <- scale_fill_brewer(palette = "Set1", name = "Treatment")
  }

  # plot
  p <- ggplot(pca_df, aes(x = PC1, y = PC2, fill = Treatment)) +
    geom_point(aes(alpha = 0.2), size = 6, shape = 21, color = "black") +
    fill_scale +
    guides(alpha = FALSE) +
    theme_classic() +
    theme(axis.title.x = element_text(size = 14),
          axis.title.y = element_text(size = 14),
          plot.title = element_text(hjust = 0.5)) +
    labs(
      title = paste0("PCA - ", stage),
      x = paste0("PC1 (", round(var_explained[1], 1), "%)"),
      y = paste0("PC2 (", round(var_explained[2], 1), "%)")
    )

  # save (adjust size/resolution or use png(...) if you prefer)
  outfile <- paste0("PCA_all_rand_", stage, ".png")
  ggsave(outfile, plot = p, width = 12, height = 7, dpi = 300)
  cat("  saved:", outfile, "\n")
}
```

### Venn Diagram for called loci
```{bash}
cut -f1,2 CASE.dp20.Block10.cov.sync > Block10.pos
cut -f1,2 CASE.dp20.Block11.cov.sync > Block11.pos
cut -f1,2 CASE.dp20.Block12.cov.sync > Block12.pos

cat <(echo "SNP") <(cat Block1*.pos | cut -f1,2 | sort | uniq -c | mawk '$1 <2' | mawk '{print $2"_"$3}') > singleton.loci
```

```{r}
# Read and preprocess data
B10.loci <- read.table("Block10.pos", header = TRUE)
colnames(B10.loci) <- c("CHROM", "BP")
B10.loci <- B10.loci %>% mutate(BLOCK_10 = TRUE)

B11.loci <- read.table("Block11.pos", header = TRUE)
colnames(B11.loci) <- c("CHROM", "BP")
B11.loci <- B11.loci %>% mutate(BLOCK_11 = TRUE)

B12.loci <- read.table("Block12.pos", header = TRUE)
colnames(B12.loci) <- c("CHROM", "BP")
B12.loci <- B12.loci %>% mutate(BLOCK_12 = TRUE)

# Full join all data frames to preserve all CHROM and BP combinations
pos_df <- B10.loci %>%
  full_join(B11.loci %>% select(CHROM, BP, BLOCK_11), by = c("CHROM", "BP")) %>%
  full_join(B12.loci %>% select(CHROM, BP, BLOCK_12), by = c("CHROM", "BP"))

# Replace NA values in the BLOCK_* columns with FALSE
pos_df <- pos_df %>%
  mutate(
    BLOCK_10 = ifelse(is.na(BLOCK_10), FALSE, BLOCK_10),
    BLOCK_11 = ifelse(is.na(BLOCK_11), FALSE, BLOCK_11),
    BLOCK_12 = ifelse(is.na(BLOCK_12), FALSE, BLOCK_12)
  )


loci.b10 <- nrow(subset(pos_df,BLOCK_10 == TRUE))
loci.b11 <- nrow(subset(pos_df,BLOCK_11 == TRUE))
loci.b12 <- nrow(subset(pos_df,BLOCK_12 == TRUE))

loci.b10.b11 <- nrow(subset(pos_df,BLOCK_10 == TRUE & BLOCK_11 ==TRUE))
loci.b10.b12 <- nrow(subset(pos_df,BLOCK_10 == TRUE & BLOCK_12 ==TRUE))
loci.b11.b12 <- nrow(subset(pos_df,BLOCK_11 == TRUE & BLOCK_12 ==TRUE))
loci.b10.b11.b12 <- nrow(subset(pos_df,BLOCK_10 == TRUE & BLOCK_11 ==TRUE & BLOCK_12 ==TRUE))

png(filename="VenSNPs.png", type="cairo",units="px", width=4000, height=4000, res=500, bg="transparent")

venn.plot <- draw.triple.venn(
  area1=loci.b10, area2=loci.b11, area3=loci.b12, 
  n12= loci.b10.b11, n13=loci.b10.b12, 
  n23=loci.b11.b12,scaled=TRUE,
  n123=loci.b10.b11.b12, cex=rep(2, 7),
  category = c("Block 10", "Block 11", "Block 12"),
  fill = cbPaletteSmall3,cat.cex = rep(2, 3),
  cat.col = cbPaletteSmall3, label.col = rep("white", 7))
dev.off()
```

# Individual Blocks for outlier testing
#### B10
```{bash, eval=FALSE}
source activate CASE

bash ../scripts/sum.sh <(cut -f1,2,3,13-16 CASE.dp20.Block10.cov.sync) > CASE.B10.IS.sync
paste CASE.B10.IS.sync <(cut -f4 CASE.B10.IS.sync) <(cut -f4 CASE.B10.IS.sync)  > CASE.B10.ISsum3.sync

AV_COV=$(mawk -f ../scripts/add_cov_sync <(cut -f13-16 --complement CASE.dp20.Block10.cov.sync) | mawk '{sum=sum+$21} END {print sum/NR}')

echo $AV_COV

mawk -f ../scripts/add_cov_sync CASE.B10.ISsum3.sync | mawk -v x=$AV_COV '$7 >= x' | mawk '!/CHR/' | cut -f1-6 > CASE.B10.ISsum3nh.sync

mawk -f ../scripts/add_cov_sync_IS CASE.B10.ISsum3.sync | mawk -v x=$AV_COV '$7 >= x'> CASE.B10.ISsum3nh.cov.sync


paste CASE.dp20.Block10.cov.sync <(mawk -f ../scripts/add_cov_sync CASE.B10.ISsum3.sync ) | mawk -v x=$AV_COV '$29 >= x' | cut -f1-22 > temp.sync

mv temp.sync CASE.dp20.Block10.COV.sync
#cp CASE.dp20.Block10.cov.sync CASE.dp20.Block10.COV.sync
mawk -f ../scripts/add_cov_sync <(cut -f13-16,20- --complement CASE.dp20.Block10.COV.sync) > temp.cov


paste temp.cov <(cut -f7 CASE.B10.ISsum3nh.cov.sync) | mawk -v OFS='\t' '{if ($18 > $19) {$18=$19; print $0} else {print $0}}' | cut -f1-18> CASE.dp20.Block10.COV
```

```{bash, eval=FALSE}
source activate random_draw

#python ../scripts/sub_sample.py CASE.dp20.Block10.COV CASE.B10.ISsum3nh.sync CASE.B10.ISs.sync
python ../scripts/sub_sample.py CASE.B10.ISsum3nh.cov.sync CASE.B10.ISsum3nh.sync CASE.B10.ISs.sync

cat <(echo -e "CHROM\tPOS\tREF\tIS_RS1\tIS_RS2\tIS_RS3") CASE.B10.ISs.sync > CASE.B10.ISsum.sync

paste <(cut -f1,2,3,4 CASE.B10.ISsum.sync) <(cut -f4 CASE.dp20.Block10.COV.sync) <(cut -f5 CASE.B10.ISsum.sync) <(cut -f5 CASE.dp20.Block10.COV.sync) <(cut -f6 CASE.B10.ISsum.sync) <(cut -f6 CASE.dp20.Block10.COV.sync)> CA.B10.sync

head -1 CA.B10.sync

mawk '!/CHR/' CA.B10.sync > CA.B10.input

paste <(cut -f1,2,3,4 CASE.B10.ISsum.sync)  <(cut -f7 CASE.dp20.Block10.COV.sync) <(cut -f5 CASE.B10.ISsum.sync) <(cut -f8 CASE.dp20.Block10.COV.sync) <(cut -f6 CASE.B10.ISsum.sync) <(cut -f9 CASE.dp20.Block10.COV.sync)> CASE.B10.sync

head -1 CASE.B10.sync
tail -n +2 CASE.B10.sync > CASE.B10.input


paste <(cut -f1,2,3,4 CASE.B10.ISsum.sync) <(cut -f17 CASE.dp20.Block10.COV.sync) <(cut -f5 CASE.B10.ISsum.sync) <(cut -f19 CASE.dp20.Block10.COV.sync) <(cut -f6 CASE.B10.ISsum.sync) <(cut -f18 CASE.dp20.Block10.COV.sync) > SE.B10.sync

head -1 SE.B10.sync
tail -n +2 SE.B10.sync > SE.B10.input


paste <(cut -f1,2,3,4 CASE.B10.ISsum.sync) <(cut -f10 CASE.dp20.Block10.COV.sync) <(cut -f5 CASE.B10.ISsum.sync) <(cut -f11 CASE.dp20.Block10.COV.sync) <(cut -f6 CASE.B10.ISsum.sync) <(cut -f12 CASE.dp20.Block10.COV.sync)> CON.B10.sync

head -1 CON.B10.sync
tail -n +2 CON.B10.sync | mawk '!/0:0:0:0:0:0/' > CON.B10.input
```

#### B11
```{bash, eval=FALSE}
AV_COV=$(mawk -f ../scripts/add_cov_sync <(cut -f16-19 --complement CASE.dp20.Block11.cov.sync) | mawk '{sum=sum+$22} END {print sum/NR}')

echo $AV_COV

bash ../scripts/sum.sh <(cut -f1,2,3,16-19 CASE.dp20.Block11.cov.sync) > CASE.B11.IS.sync
paste CASE.B11.IS.sync <(cut -f4 CASE.B11.IS.sync) <(cut -f4 CASE.B11.IS.sync) <(cut -f4 CASE.B11.IS.sync) > CASE.B11.ISsum4.sync

mawk -f ../scripts/add_cov_sync CASE.B11.ISsum4.sync | mawk -v x=$AV_COV '$8 >= x' | mawk '!/CHR/' | cut -f1-7 > CASE.B11.ISsum4nh.sync

paste CASE.dp20.Block11.cov.sync <(mawk -f ../scripts/add_cov_sync CASE.B11.ISsum4.sync ) | mawk -v x=$AV_COV '$34 >= x' | cut -f1-23 > temp.sync

mv temp.sync CASE.dp20.Block11.COV.sync

mawk -f ../scripts/add_cov_sync_IS CASE.B11.ISsum4.sync | mawk -v x=$AV_COV '$8 >= x'  > CASE.B11.ISsum4.cov.sync
mawk -f ../scripts/add_cov_sync <(cut -f16-19 --complement CASE.dp20.Block11.COV.sync) > CASE.dp20.Block11.COV

source activate random_draw

#python ../scripts/sub_sample.py CASE.dp20.Block11.COV CASE.B11.ISsum4nh.sync CASE.B11.ISs.sync
python ../scripts/sub_sample.py CASE.B11.ISsum4.cov.sync CASE.B11.ISsum4nh.sync CASE.B11.ISs.sync

cat <(echo -e "CHROM\tPOS\tREF\tIS_RS1\tIS_RS2\tIS_RS3\tIS_R4") CASE.B11.ISs.sync > CASE.B11.ISsum.sync

paste <(cut -f1,2,3,4 CASE.B11.ISsum.sync) <(cut -f4 CASE.dp20.Block11.COV.sync) <(cut -f5 CASE.B11.ISsum.sync) <(cut -f 5 CASE.dp20.Block11.COV.sync) <(cut -f6 CASE.B11.ISsum.sync) <(cut -f6 CASE.dp20.Block11.COV.sync) <(cut -f7 CASE.B11.ISsum.sync) <(cut -f7 CASE.dp20.Block11.COV.sync) > CA.B11.sync

head -1 CA.B11.sync
tail -n +2 CA.B11.sync > CA.B11.input

paste <(cut -f1,2,3,4 CASE.B11.ISsum.sync) <(cut -f8 CASE.dp20.Block11.COV.sync) <(cut -f5 CASE.B11.ISsum.sync) <(cut -f 9 CASE.dp20.Block11.COV.sync) <(cut -f6 CASE.B11.ISsum.sync) <(cut -f10 CASE.dp20.Block11.COV.sync) <(cut -f7 CASE.B11.ISsum.sync) <(cut -f11 CASE.dp20.Block11.COV.sync) > CASE.B11.sync

head -1 CASE.B11.sync
tail -n +2 CASE.B11.sync > CASE.B11.input

paste <(cut -f1,2,3,4 CASE.B11.ISsum.sync) <(cut -f20 CASE.dp20.Block11.COV.sync) <(cut -f5 CASE.B11.ISsum.sync) <(cut -f 21 CASE.dp20.Block11.COV.sync) <(cut -f6 CASE.B11.ISsum.sync) <(cut -f22 CASE.dp20.Block11.COV.sync) <(cut -f7 CASE.B11.ISsum.sync) <(cut -f23 CASE.dp20.Block11.COV.sync) > SE.B11.sync

head -1 SE.B11.sync
tail -n +2 SE.B11.sync > SE.B11.input

paste <(cut -f1,2,3,4 CASE.B11.ISsum.sync) <(cut -f12 CASE.dp20.Block11.COV.sync) <(cut -f5 CASE.B11.ISsum.sync) <(cut -f 13 CASE.dp20.Block11.COV.sync) <(cut -f6 CASE.B11.ISsum.sync) <(cut -f14 CASE.dp20.Block11.COV.sync) <(cut -f7 CASE.B11.ISsum.sync) <(cut -f15 CASE.dp20.Block11.COV.sync) > CON.B11.sync

head -1 CON.B11.sync
tail -n +2 CON.B11.sync > CON.B11.input
```
#### B12


```{bash, eval=FALSE}
source activate CASE

bash ../scripts/sum.sh <(cut -f1,2,3,16-19 CASE.dp20.Block12.cov.sync) > CASE.B12.IS.sync
paste CASE.B12.IS.sync <(cut -f4 CASE.B12.IS.sync) <(cut -f4 CASE.B12.IS.sync) <(cut -f4 CASE.B12.IS.sync) > CASE.B12.ISsum4.sync

AV_COV=$(mawk -f ../scripts/add_cov_sync <(cut -f16-19 --complement CASE.dp20.Block12.cov.sync) | mawk '{sum=sum+$22} END {print sum/NR}')

echo $AV_COV

mawk -f ../scripts/add_cov_sync CASE.B12.ISsum4.sync | mawk -v x=$AV_COV '$8 >= x' | mawk '!/CHR/' | cut -f1-7 > CASE.B12.ISsum4nh.sync

paste CASE.dp20.Block12.cov.sync <(mawk -f ../scripts/add_cov_sync CASE.B12.ISsum4.sync ) | mawk -v x=$AV_COV '$34 >= x' | cut -f1-26 > temp.sync

mv -f temp.sync CASE.dp20.Block12.COV.sync

mawk -f ../scripts/add_cov_sync_IS CASE.B12.ISsum4.sync | mawk -v x=$AV_COV '$8 >= x'  > CASE.B12.ISsum4nh.cov.sync

mawk -f ../scripts/add_cov_sync <(cut -f16-19,24- --complement CASE.dp20.Block12.COV.sync) > CASE.dp20.Block12.COV


paste CASE.dp20.Block12.COV <(cut -f8 CASE.B12.ISsum4nh.cov.sync) | mawk -v OFS='\t' '{if ($22 > $23) {$22=$23; print $0} else {print $0}}' > temp.cov.sync

cut -f1-22 temp.cov.sync > CASE.dp20.Block12.COV
rm temp.cov.sync

```

```{bash, eval=FALSE}
source activate random_draw

#python ../scripts/sub_sample.py CASE.dp20.Block12.COV CASE.B12.ISsum4nh.sync CASE.B12.ISs.sync
python ../scripts/sub_sample.py CASE.B12.ISsum4nh.cov.sync CASE.B12.ISsum4nh.sync CASE.B12.ISs.sync

cat <(echo -e "CHROM\tPOS\tREF\tIS_RS1\tIS_RS2\tIS_RS3\tIS_R4") CASE.B12.ISs.sync > CASE.B12.ISsum.sync

paste <(cut -f1,2,3,4 CASE.B12.ISsum.sync) <(cut -f4 CASE.dp20.Block12.COV.sync) <(cut -f5 CASE.B12.ISsum.sync) <(cut -f 5 CASE.dp20.Block12.COV.sync) <(cut -f6 CASE.B12.ISsum.sync) <(cut -f6 CASE.dp20.Block12.COV.sync) <(cut -f7 CASE.B12.ISsum.sync) <(cut -f7 CASE.dp20.Block12.COV.sync) > CA.B12.sync

head -1 CA.B12.sync
tail -n +2 CA.B12.sync > CA.B12.input

paste <(cut -f1,2,3,4 CASE.B12.ISsum.sync) <(cut -f8 CASE.dp20.Block12.COV.sync) <(cut -f5 CASE.B12.ISsum.sync) <(cut -f 9 CASE.dp20.Block12.COV.sync) <(cut -f6 CASE.B12.ISsum.sync) <(cut -f10 CASE.dp20.Block12.COV.sync) <(cut -f7 CASE.B12.ISsum.sync) <(cut -f11 CASE.dp20.Block12.COV.sync) > CASE.B12.sync

head -1 CASE.B12.sync
tail -n +2 CASE.B12.sync > CASE.B12.input

paste <(cut -f1,2,3,4 CASE.B12.ISsum.sync) <(cut -f20 CASE.dp20.Block12.COV.sync) <(cut -f5 CASE.B12.ISsum.sync) <(cut -f 21 CASE.dp20.Block12.COV.sync) <(cut -f6 CASE.B12.ISsum.sync) <(cut -f22 CASE.dp20.Block12.COV.sync) <(cut -f7 CASE.B12.ISsum.sync) <(cut -f23 CASE.dp20.Block12.COV.sync) > SE.B12.sync

head -1 SE.B12.sync
tail -n +2 SE.B12.sync > SE.B12.input

paste <(cut -f1,2,3,4 CASE.B12.ISsum.sync) <(cut -f12 CASE.dp20.Block12.COV.sync) <(cut -f5 CASE.B12.ISsum.sync) <(cut -f 13 CASE.dp20.Block12.COV.sync) <(cut -f6 CASE.B12.ISsum.sync) <(cut -f14 CASE.dp20.Block12.COV.sync) <(cut -f7 CASE.B12.ISsum.sync) <(cut -f15 CASE.dp20.Block12.COV.sync) > CON.B12.sync

head -1 CON.B12.sync
tail -n +2 CON.B12.sync > CON.B12.input

```
### Calculate P-values

#### Modified CMH
```{r}
ca.b10.sync <- read.sync(file="CA.B10.input", gen=c(0, 1, 0, 1, 0, 1), repl=c(1, 1, 2, 2, 3,3))
ca.b10.cov <- coverage(ca.b10.sync,gen=c(0, 1, 0, 1, 0, 1), repl=c(1, 1, 2, 2, 3,3))
ca.b10.af <- af(ca.b10.sync,gen=c(0, 1, 0, 1, 0, 1), repl=c(1, 1, 2, 2, 3,3))

case.b10.sync <- read.sync(file="CASE.B10.input", gen=c(0, 1, 0, 1, 0, 1), repl=c(1, 1, 2, 2, 3,3))
case.b10.cov <- coverage(case.b10.sync,gen=c(0, 1, 0, 1, 0, 1), repl=c(1, 1, 2, 2, 3,3))
case.b10.af <- af(case.b10.sync,gen=c(0, 1, 0, 1, 0, 1), repl=c(1, 1, 2, 2, 3,3))

se.b10.sync <- read.sync(file="SE.B10.input", gen=c(0, 1, 0, 1, 0, 1), repl=c(1, 1, 2, 2, 3,3))
se.b10.cov <- coverage(se.b10.sync,gen=c(0, 1, 0, 1, 0, 1), repl=c(1, 1, 2, 2, 3,3))
se.b10.af <- af(se.b10.sync,gen=c(0, 1, 0, 1, 0, 1), repl=c(1, 1, 2, 2, 3,3))

con.b10.sync <- read.sync(file="CON.B10.input", gen=c(0, 1, 0, 1, 0, 1), repl=c(1, 1, 2, 2, 3,3))
con.b10.cov <- coverage(con.b10.sync,gen=c(0, 1, 0, 1, 0, 1), repl=c(1, 1, 2, 2, 3,3))
con.b10.af <- af(con.b10.sync,gen=c(0, 1, 0, 1, 0, 1), repl=c(1, 1, 2, 2, 3,3))
```

```{r}
ca.b10.pvals <- adapted.cmh.test(freq=ca.b10.af, coverage=ca.b10.cov, Ne=rep(250, 3), gen=c(0, 1, 0, 1, 0, 1), repl=c(1, 1, 2, 2, 3,3), poolSize=rep(c(40000,10000), ncol(ca.b10.af)/2), MeanStart = TRUE, mincov =20)

pcadf <- as.data.table(cbind(row.names(ca.b10.af),ca.b10.pvals))
colnames(pcadf) <- c("Locus","PVAL")
pcadf$GROUP <- "PCA"


case.b10.pvals <- adapted.cmh.test(freq=case.b10.af, coverage=case.b10.cov, Ne=rep(250, 3), gen=c(0, 1, 0, 1, 0, 1), repl=c(1, 1, 2, 2, 3,3), poolSize=rep(c(40000,10000), ncol(case.b10.af)/2), MeanStart = TRUE, mincov =20)

pcasedf <- as.data.table(cbind(row.names(case.b10.af),case.b10.pvals))
colnames(pcasedf) <- c("Locus","PVAL")
pcasedf$GROUP <- "PCASE"


se.b10.pvals <- adapted.cmh.test(freq=se.b10.af, coverage=se.b10.cov, Ne=rep(250, 3), gen=c(0, 1, 0, 1, 0, 1), repl=c(1, 1, 2, 2, 3,3), poolSize=rep(c(40000,10000), ncol(se.b10.af)/2), MeanStart = TRUE, mincov =20)

psedf <- as.data.table(cbind(row.names(se.b10.af),se.b10.pvals))
colnames(psedf) <- c("Locus","PVAL")
psedf$GROUP <- "PSE"


con.b10.pvals <- adapted.cmh.test(freq=con.b10.af, coverage=con.b10.cov, Ne=rep(250, 3), gen=c(0, 1, 0, 1, 0, 1), repl=c(1, 1, 2, 2, 3,3), poolSize=rep(c(40000,10000), ncol(con.b10.af)/2), MeanStart = TRUE, mincov =20)

pcondf <- as.data.table(cbind(row.names(con.b10.af),con.b10.pvals))
colnames(pcondf) <- c("Locus","PVAL")
pcondf$GROUP <- "PCON"

B10.pval.table <- bind_rows(pcadf,pcasedf,pcondf,psedf)
B10.pval.table <- B10.pval.table %>%
  separate(Locus, into = c("CHROM", "BP"), sep = "\\.(?=[^.]+$)")
B10.pval.table$PVAL <- as.numeric(B10.pval.table$PVAL)
```

```{r}
ca.b11.sync <- read.sync(file="CA.B11.input", gen=c(0, 1, 0, 1, 0, 1, 0, 1), repl=c(1, 1, 2, 2, 3, 3, 4 , 4 ))
ca.b11.cov <- coverage(ca.b11.sync,gen=c(0, 1, 0, 1, 0, 1, 0, 1), repl=c(1, 1, 2, 2, 3, 3, 4 , 4 ))
ca.b11.af <- af(ca.b11.sync,gen=c(0, 1, 0, 1, 0, 1, 0, 1), repl=c(1, 1, 2, 2, 3, 3, 4 , 4 ))

case.b11.sync <- read.sync(file="CASE.B11.input", gen=c(0, 1, 0, 1, 0, 1, 0, 1), repl=c(1, 1, 2, 2, 3, 3, 4 , 4 ))
case.b11.cov <- coverage(case.b11.sync,gen=c(0, 1, 0, 1, 0, 1, 0, 1), repl=c(1, 1, 2, 2, 3, 3, 4 , 4 ))
case.b11.af <- af(case.b11.sync,gen=c(0, 1, 0, 1, 0, 1, 0, 1), repl=c(1, 1, 2, 2, 3, 3, 4 , 4 ))

se.b11.sync <- read.sync(file="SE.B11.input", gen=c(0, 1, 0, 1, 0, 1, 0, 1), repl=c(1, 1, 2, 2, 3, 3, 4 , 4 ))
se.b11.cov <- coverage(se.b11.sync,gen=c(0, 1, 0, 1, 0, 1, 0, 1), repl=c(1, 1, 2, 2, 3, 3, 4 , 4 ))
se.b11.af <- af(se.b11.sync,gen=c(0, 1, 0, 1, 0, 1, 0, 1), repl=c(1, 1, 2, 2, 3, 3, 4 , 4 ))

con.b11.sync <- read.sync(file="CON.B11.input", gen=c(0, 1, 0, 1, 0, 1, 0, 1), repl=c(1, 1, 2, 2, 3, 3, 4 , 4 ))
con.b11.cov <- coverage(con.b11.sync,gen=c(0, 1, 0, 1, 0, 1, 0, 1), repl=c(1, 1, 2, 2, 3, 3, 4 , 4 ))
con.b11.af <- af(con.b11.sync,gen=c(0, 1, 0, 1, 0, 1, 0, 1), repl=c(1, 1, 2, 2, 3, 3, 4 , 4 ))
```

```{r}
ca.b11.pvals <- adapted.cmh.test(freq=ca.b11.af, coverage=ca.b11.cov, Ne=rep(1000, 4), gen=c(0, 1, 0, 1, 0, 1, 0, 1), repl=c(1, 1, 2, 2, 3, 3, 4, 4), poolSize=rep(c(40000,10000), ncol(ca.b11.af)/2), MeanStart = TRUE, mincov =14)

pcadf <- as.data.table(cbind(row.names(ca.b11.af),ca.b11.pvals))
colnames(pcadf) <- c("Locus","PVAL")
pcadf$GROUP <- "PCA"

case.b11.pvals <- adapted.cmh.test(freq=case.b11.af, coverage=case.b11.cov, Ne=rep(1000, 4), gen=c(0, 1, 0, 1, 0, 1, 0, 1), repl=c(1, 1, 2, 2, 3, 3, 4, 4), poolSize=rep(c(40000,10000), ncol(case.b11.af)/2), MeanStart = TRUE, mincov =14)

pcasedf <- as.data.table(cbind(row.names(case.b11.af),case.b11.pvals))
colnames(pcasedf) <- c("Locus","PVAL")
pcasedf$GROUP <- "PCASE"

se.b11.pvals <- adapted.cmh.test(freq=se.b11.af, coverage=se.b11.cov, Ne=rep(1000, 4), gen=c(0, 1, 0, 1, 0, 1, 0, 1), repl=c(1, 1, 2, 2, 3, 3, 4, 4), poolSize=rep(c(40000,10000), ncol(se.b11.af)/2), MeanStart = TRUE, mincov =14)

psedf <- as.data.table(cbind(row.names(se.b11.af),se.b11.pvals))
colnames(psedf) <- c("Locus","PVAL")
psedf$GROUP <- "PSE"

con.b11.pvals <- adapted.cmh.test(freq=con.b11.af, coverage=con.b11.cov, Ne=rep(1000, 4), gen=c(0, 1, 0, 1, 0, 1, 0, 1), repl=c(1, 1, 2, 2, 3, 3, 4, 4), poolSize=rep(c(40000,10000), ncol(con.b11.af)/2), MeanStart = TRUE, mincov =14)

pcondf <- as.data.table(cbind(row.names(con.b11.af),con.b11.pvals))
colnames(pcondf) <- c("Locus","PVAL")
pcondf$GROUP <- "PCON"


B11.pval.table <- bind_rows(pcadf,pcasedf,pcondf,psedf)
B11.pval.table <- B11.pval.table %>%
  separate(Locus, into = c("CHROM", "BP"), sep = "\\.(?=[^.]+$)")
B11.pval.table$PVAL <- as.numeric(B11.pval.table$PVAL)
```

```{r}
ca.b12.sync <- read.sync(file="CA.B12.input", gen=c(0, 1, 0, 1, 0, 1, 0, 1), repl=c(1, 1, 2, 2, 3, 3, 4 , 4 ))
ca.b12.cov <- coverage(ca.b12.sync,gen=c(0, 1, 0, 1, 0, 1, 0, 1), repl=c(1, 1, 2, 2, 3, 3, 4 , 4 ))
ca.b12.af <- af(ca.b12.sync,gen=c(0, 1, 0, 1, 0, 1, 0, 1), repl=c(1, 1, 2, 2, 3, 3, 4 , 4 ))

case.b12.sync <- read.sync(file="CASE.B12.input", gen=c(0, 1, 0, 1, 0, 1, 0, 1), repl=c(1, 1, 2, 2, 3, 3, 4 , 4 ))
case.b12.cov <- coverage(case.b12.sync,gen=c(0, 1, 0, 1, 0, 1, 0, 1), repl=c(1, 1, 2, 2, 3, 3, 4 , 4 ))
case.b12.af <- af(case.b12.sync,gen=c(0, 1, 0, 1, 0, 1, 0, 1), repl=c(1, 1, 2, 2, 3, 3, 4 , 4 ))

se.b12.sync <- read.sync(file="SE.B12.input", gen=c(0, 1, 0, 1, 0, 1, 0, 1), repl=c(1, 1, 2, 2, 3, 3, 4 , 4 ))
se.b12.cov <- coverage(se.b12.sync,gen=c(0, 1, 0, 1, 0, 1, 0, 1), repl=c(1, 1, 2, 2, 3, 3, 4 , 4 ))
se.b12.af <- af(se.b12.sync,gen=c(0, 1, 0, 1, 0, 1, 0, 1), repl=c(1, 1, 2, 2, 3, 3, 4 , 4 ))

con.b12.sync <- read.sync(file="CON.B12.input", gen=c(0, 1, 0, 1, 0, 1, 0, 1), repl=c(1, 1, 2, 2, 3, 3, 4 , 4 ))
con.b12.cov <- coverage(con.b12.sync,gen=c(0, 1, 0, 1, 0, 1, 0, 1), repl=c(1, 1, 2, 2, 3, 3, 4 , 4 ))
con.b12.af <- af(con.b12.sync,gen=c(0, 1, 0, 1, 0, 1, 0, 1), repl=c(1, 1, 2, 2, 3, 3, 4 , 4 ))
```
```{r}
ca.b12.pvals <- adapted.cmh.test(freq=ca.b12.af, coverage=ca.b12.cov, Ne=rep(1000, 4), gen=c(0, 1, 0, 1, 0, 1, 0, 1), repl=c(1, 1, 2, 2, 3, 3, 4, 4), poolSize=rep(c(40000,10000), ncol(ca.b12.af)/2), MeanStart = TRUE, mincov =20)

pcadf <- as.data.table(cbind(row.names(ca.b12.af),ca.b12.pvals))
colnames(pcadf) <- c("Locus","PVAL")
pcadf$GROUP <- "PCA"

case.b12.pvals <- adapted.cmh.test(freq=case.b12.af, coverage=case.b12.cov, Ne=rep(1000, 4), gen=c(0, 1, 0, 1, 0, 1, 0, 1), repl=c(1, 1, 2, 2, 3, 3, 4, 4), poolSize=rep(c(40000,10000), ncol(case.b12.af)/2), MeanStart = TRUE, mincov =20)

pcasedf <- as.data.table(cbind(row.names(case.b12.af),case.b12.pvals))
colnames(pcasedf) <- c("Locus","PVAL")
pcasedf$GROUP <- "PCASE"

se.b12.pvals <- adapted.cmh.test(freq=se.b12.af, coverage=se.b12.cov, Ne=rep(1000, 4), gen=c(0, 1, 0, 1, 0, 1, 0, 1), repl=c(1, 1, 2, 2, 3, 3, 4, 4), poolSize=rep(c(40000,10000), ncol(se.b12.af)/2), MeanStart = TRUE, mincov =20)

psedf <- as.data.table(cbind(row.names(se.b12.af),se.b12.pvals))
colnames(psedf) <- c("Locus","PVAL")
psedf$GROUP <- "PSE"



con.b12.pvals <- adapted.cmh.test(freq=con.b12.af, coverage=con.b12.cov, Ne=rep(1000, 4), gen=c(0, 1, 0, 1, 0, 1, 0, 1), repl=c(1, 1, 2, 2, 3, 3, 4, 4), poolSize=rep(c(40000,10000), ncol(con.b12.af)/2), MeanStart = TRUE, mincov =20)

pcondf <- as.data.table(cbind(row.names(con.b12.af),con.b12.pvals))
colnames(pcondf) <- c("Locus","PVAL")
pcondf$GROUP <- "PCON"

B12.pval.table <- bind_rows(pcadf,pcasedf,pcondf,psedf)
B12.pval.table <- B12.pval.table %>%
  separate(Locus, into = c("CHROM", "BP"), sep = "\\.(?=[^.]+$)")
B12.pval.table$PVAL <- as.numeric(B12.pval.table$PVAL)
```



```{r}
Qvalue_convert(B10.pval.table) ->B10.pv
Qvalue_convert(B11.pval.table) ->B11.pv
Qvalue_convert(B12.pval.table) ->B12.pv
```

```{r}
pp1 <- rbind(B10.pv,B11.pv,B12.pv)

alpha = 0.1
B11.alpha = alpha * 1
alpha2 =0.1

ppsig.B10.1 <- Significat_subset(B10.pv,alpha,alpha2)
ppsig.B11.1 <- Significat_subset(B11.pv,B11.alpha,alpha2)
ppsig.B12.1 <- Significat_subset(B12.pv,alpha,alpha2)

ppsig.B10.1$BLOCK <- 10
ppsig.B11.1$BLOCK <- 11
ppsig.B12.1$BLOCK <- 12

ppsig3.FDR10 <- bind_rows(ppsig.B10.1,ppsig.B11.1,ppsig.B12.1)

all_sig.CA<- subset(ppsig3.FDR10, Sig.CA == "TRUE") %>% group_by(SNP) %>% filter(n()>2)
all_sig.CA$Sig.CASE<- FALSE
all_sig.CA$Sig.SE<- FALSE

all_sig.CASE<- subset(ppsig3.FDR10, Sig.CASE == "TRUE") %>% group_by(SNP) %>% filter(n()>2)
all_sig.CASE$Sig.CA<- FALSE
all_sig.CASE$Sig.SE<- FALSE

all_sig.SE<- subset(ppsig3.FDR10, Sig.SE == "TRUE") %>% group_by(SNP) %>% filter(n()>2)
all_sig.SE$Sig.CASE<- FALSE
all_sig.SE$Sig.CA<- FALSE

#all_sig <- ppsig3.FDR10 %>% group_by(SNP) %>% filter(n()>2)
all_sig <- bind_rows(all_sig.CA,all_sig.CASE,all_sig.SE)

write.table(all_sig, "Sig.Loci.3", sep="\t", row.names = FALSE, quote = FALSE)

alpha = 0.05
B11.alpha = alpha * 2
alpha2 =0.1

ppsig.B10.05 <- Significat_subset(B10.pv,alpha,alpha2)
ppsig.B11.05 <- Significat_subset(B11.pv,B11.alpha,alpha2)
ppsig.B12.05 <- Significat_subset(B12.pv,alpha,alpha2)

ppsig.B10.05$BLOCK <- 10
ppsig.B11.05$BLOCK <- 11
ppsig.B12.05$BLOCK <- 12

ppsig3.FDR05 <- rbind(ppsig.B10.05,ppsig.B11.05,ppsig.B12.05)

multi_sig.CA<- subset(ppsig3.FDR05, Sig.CA == "TRUE") %>% group_by(SNP) %>% filter(n()>1)
multi_sig.CA$Sig.CASE <- FALSE
multi_sig.CA$Sig.SE <- FALSE

multi_sig.CASE<- subset(ppsig3.FDR05, Sig.CASE == "TRUE") %>% group_by(SNP) %>% filter(n()>1)
multi_sig.CASE$Sig.CA <- FALSE
multi_sig.CASE$Sig.SE <- FALSE

multi_sig.SE<- subset(ppsig3.FDR05, Sig.SE == "TRUE") %>% group_by(SNP) %>% filter(n()>1)
multi_sig.SE$Sig.CA <- FALSE
multi_sig.SE$Sig.CASE <- FALSE

multi_sig <- rbind(multi_sig.CA,multi_sig.CASE,multi_sig.SE)

#multi_sig.CA<- subset(ppsig3.FDR05, QCA < B11.alpha) %>% group_by(SNP) %>% filter(n()>1)
#multi_sig.CASE<- subset(ppsig3.FDR05, QCASE < B11.alpha) %>% group_by(SNP) %>% filter(n()>1)
#multi_sig.SE<- subset(ppsig3.FDR05, QSE < B11.alpha) %>% group_by(SNP) %>% filter(n()>1)

#multi_sig <- rbind(ppsig3.FDR05 %>% group_by(SNP) %>% filter(n()>1))
#multi_sig <- rbind(multi_sig.CA,multi_sig.CASE,multi_sig.SE)

write.table(multi_sig, "Sig.Loci.2", sep="\t", row.names = FALSE, quote = FALSE)

alpha = 0.01
alpha2 =0.1
B11.alpha = alpha *2

ppsig.B10.01 <- Significat_subset(B10.pv,alpha,alpha2)
ppsig.B11.01 <- Significat_subset(B11.pv,B11.alpha,alpha2)
ppsig.B12.01 <- Significat_subset(B12.pv,alpha,alpha2)

conpp1 <- subset(pp1, QCON < alpha)

ppsig.B10.01$BLOCK <- 10
ppsig.B11.01$BLOCK <- 11
ppsig.B12.01$BLOCK <- 12

ppsig3.FDR01 <- rbind(ppsig.B10.01,ppsig.B11.01,ppsig.B12.01)

write.table(ppsig3.FDR01, "Sig.Loci.FDR01.1", sep="\t", row.names = FALSE, quote = FALSE)
write.table(ppsig.B10.01, "Sig.Loci.FDR01.Block10", sep="\t", row.names = FALSE, quote = FALSE)
write.table(ppsig.B11.01, "Sig.Loci.FDR01.Block11", sep="\t", row.names = FALSE, quote = FALSE)
write.table(ppsig.B12.01, "Sig.Loci.FDR01.Block12", sep="\t", row.names = FALSE, quote = FALSE)

multi_sig.CA.01<- subset(ppsig3.FDR01, Sig.CA == "TRUE") %>% group_by(SNP) %>% filter(n()>1)
multi_sig.CA.01$Sig.CASE <- FALSE
multi_sig.CA.01$Sig.SE <- FALSE

multi_sig.CASE.01<- subset(ppsig3.FDR01, Sig.CASE == "TRUE") %>% group_by(SNP) %>% filter(n()>1)
multi_sig.CASE.01$Sig.CA <- FALSE
multi_sig.CASE.01$Sig.SE <- FALSE

multi_sig.SE.01<- subset(ppsig3.FDR01, Sig.SE == "TRUE") %>% group_by(SNP) %>% filter(n()>1)
multi_sig.SE.01$Sig.CA <- FALSE
multi_sig.SE.01$Sig.CASE <- FALSE

multi_sig.01 <- rbind(multi_sig.CA.01,multi_sig.CASE.01,multi_sig.SE.01)

write.table(multi_sig.01, "Sig.Loci.FDR01.2", sep="\t", row.names = FALSE, quote = FALSE)

alpha = 0.01
alpha2 =0.1
B11.alpha = alpha * 2


ppsig.B10 <- Significat_subset(B10.pv,alpha,alpha2)
ppsig.B10 <- subset(ppsig.B10, Sig.CASE == TRUE & QCASE < quantile(ppsig.B10$QCASE, na.rm = TRUE, probs = 0.01)
                    | Sig.CA == TRUE & QCA < quantile(ppsig.B10$QCA, na.rm = TRUE, probs = 0.01)
                    | Sig.SE == TRUE & QSE < quantile(ppsig.B10$QSE, na.rm = TRUE, probs = 0.01)
                    )
ppsig.B11 <- Significat_subset(B11.pv,B11.alpha,alpha2)
ppsig.B11 <- subset(ppsig.B11, Sig.CASE == TRUE & QCASE < quantile(ppsig.B11$QCASE, na.rm = TRUE, probs = 0.01)
                    | Sig.CA == TRUE & QCA < quantile(ppsig.B11$QCA, na.rm = TRUE, probs = 0.01)
                    | Sig.SE == TRUE & QSE < quantile(ppsig.B11$QSE, na.rm = TRUE, probs = 0.01)
                    )
ppsig.B12 <- Significat_subset(B12.pv,alpha,alpha2)
ppsig.B12 <- subset(ppsig.B12, Sig.CASE == TRUE & QCASE < quantile(ppsig.B12$QCASE, na.rm = TRUE, probs = 0.01)
                    | Sig.CA == TRUE & QCA < quantile(ppsig.B12$QCA, na.rm = TRUE, probs = 0.01)
                    | Sig.SE == TRUE & QSE < quantile(ppsig.B12$QSE, na.rm = TRUE, probs = 0.01)
                    )

ppsig.B10$BLOCK <- 10
ppsig.B11$BLOCK <- 11
ppsig.B12$BLOCK <- 12

singleton <- read.table("singleton.loci",header=TRUE)

ppsig1.s <- semi_join(rbind(ppsig.B10,ppsig.B11,ppsig.B12),singleton, by = "SNP")


```

# Across all blocks
#### Create sync file
```{bash, eval=FALSE}
source activate CASE

head -1 CASE.All.Blocks.cov.sync > h.temp
tail -n +2 CASE.All.Blocks.cov.sync | sort -k1,1 -k2,2n > b.temp

cat h.temp b.temp > CASE.All.Blocks.cov.s.sync
rm h.temp b.temp
cp CASE.All.Blocks.cov.sync CASE.All.Blocks.cov.s.sync

bash ../scripts/sum.sh <(cut -f1,2,3,37,39,41,43 CASE.All.Blocks.cov.s.sync) > CASE.B10.AB.ISsum.sync
bash ../scripts/sum.sh <(cut -f1,2,3,38,40,42,44 CASE.All.Blocks.cov.s.sync) > CASE.B11.AB.ISsum.sync
bash ../scripts/sum.sh <(cut -f1,2,3,45-48 CASE.All.Blocks.cov.s.sync) > CASE.B12.AB.ISsum.sync


paste CASE.All.Blocks.cov.s.sync <(mawk -f ../scripts/add_cov_sync CASE.B10.AB.ISsum.sync | cut -f5 ) <(mawk -f ../scripts/add_cov_sync CASE.B11.AB.ISsum.sync | cut -f 5) <(mawk -f ../scripts/add_cov_sync CASE.B12.AB.ISsum.sync | cut -f5) | mawk '$63 >72 && $64 > 72 && $65 > 72 && $61 > 1' | cut -f1-62 > CASE.All.Blocks.cov.fil.sync

bash ../scripts/sum.sh <(cut -f1,2,3,37,39,41,43 CASE.All.Blocks.cov.fil.sync) > CASE.B10.AB.ISsum.sync
bash ../scripts/sum.sh <(cut -f1,2,3,38,40,42,44 CASE.All.Blocks.cov.fil.sync) > CASE.B11.AB.ISsum.sync
bash ../scripts/sum.sh <(cut -f1,2,3,45-48 CASE.All.Blocks.cov.fil.sync) > CASE.B12.AB.ISsum.sync

mawk -f ../scripts/add_cov_sync <(cut -f1-3,7,9,12,15,19,22,26,30,32,51,52,55  CASE.All.Blocks.cov.fil.sync ) > B10.AB.cov.sync
mawk -f ../scripts/add_cov_sync <(cut -f1-3,5,10,14,16,23,24,27,29,35,49,54,58 CASE.All.Blocks.cov.fil.sync) > B11.AB.cov.sync
#mawk -f ../scripts/add_cov_sync <(cut -f1-3,5,8,10,14,16,21,23,24,27,29,33,35,49,53,54,58 CASE.All.Blocks.cov.fil.sync) > B11.AB.cov.sync

mawk -f ../scripts/add_cov_sync <(cut -f1-3,4,6,13,17,18,20,28,31,34,56,57,59 CASE.All.Blocks.cov.fil.sync) > B12.AB.cov.sync
#mawk -f ../scripts/add_cov_sync <(cut -f1-3,4,6,11,13,17,18,20,25,28,31,34,36,50,56,57,59 CASE.All.Blocks.cov.fil.sync) > B12.AB.cov.sync

#paste CASE.B10.AB.ISsum.sync <(cut -f4 CASE.B10.AB.ISsum.sync) <(cut -f4 CASE.B10.AB.ISsum.sync)   > CASE.B10.AB.ISsum3.sync

#paste CASE.B11.AB.ISsum.sync <(cut -f4 CASE.B11.AB.ISsum.sync) <(cut -f4 CASE.B11.AB.ISsum.sync)  > CASE.B11.AB.ISsum4.sync

#paste CASE.B12.AB.ISsum.sync <(cut -f4 CASE.B12.AB.ISsum.sync) <(cut -f4 CASE.B12.AB.ISsum.sync)  > CASE.B12.AB.ISsum4.sync

paste CASE.B10.AB.ISsum.sync <(cut -f4 CASE.B10.AB.ISsum.sync) <(cut -f4 CASE.B10.AB.ISsum.sync) <(cut -f4 CASE.B10.AB.ISsum.sync)  > CASE.B10.AB.ISsum3.sync
paste CASE.B11.AB.ISsum.sync <(cut -f4 CASE.B11.AB.ISsum.sync) <(cut -f4 CASE.B11.AB.ISsum.sync) <(cut -f4 CASE.B11.AB.ISsum.sync)  > CASE.B11.AB.ISsum4.sync
paste CASE.B12.AB.ISsum.sync <(cut -f4 CASE.B12.AB.ISsum.sync) <(cut -f4 CASE.B12.AB.ISsum.sync) <(cut -f4 CASE.B12.AB.ISsum.sync)  > CASE.B12.AB.ISsum4.sync


mawk -f ../scripts/add_cov_sync_IS CASE.B10.AB.ISsum3.sync  > CASE.B10.AB.ISsum3.cov.sync
mawk -f ../scripts/add_cov_sync_IS CASE.B11.AB.ISsum4.sync  > CASE.B11.AB.ISsum4.cov.sync
mawk -f ../scripts/add_cov_sync_IS CASE.B12.AB.ISsum4.sync  > CASE.B12.AB.ISsum4.cov.sync

paste B10.AB.cov.sync <(cut -f7 CASE.B10.AB.ISsum3.cov.sync) | mawk -v OFS='\t' '{if (NR > 1 && $19 > $20) {$19=$20; print $0} else {print $0}}' > temp.cov.sync
cut -f1-18 temp.cov.sync > B10.AB.cov.sync

paste B11.AB.cov.sync <(cut -f7 CASE.B11.AB.ISsum4.cov.sync) | mawk -v OFS='\t' '{if (NR > 1 && $19 > $20) {$19=$20; print $0} else {print $0}}' > temp.cov.sync
cut -f1-18 temp.cov.sync > B11.AB.cov.sync

paste B12.AB.cov.sync <(cut -f7 CASE.B12.AB.ISsum4.cov.sync) | mawk -v OFS='\t' '{if (NR > 1 && $19 > $20) {$19=$20; print $0} else {print $0}}' > temp.cov.sync
cut -f1-18 temp.cov.sync > B12.AB.cov.sync
rm temp.cov.sync

```

```{bash, eval=FALSE}
source activate random_draw
python ../scripts/sub_sample.py CASE.B10.AB.ISsum3.cov.sync <( mawk '!/CHR/' CASE.B10.AB.ISsum3.sync) CASE.B10.ISs.AB.sync
python ../scripts/sub_sample.py CASE.B11.AB.ISsum4.cov.sync <( mawk '!/CHR/' CASE.B11.AB.ISsum4.sync) CASE.B11.ISs.AB.sync
python ../scripts/sub_sample.py CASE.B12.AB.ISsum4.cov.sync <(mawk '!/CHR/' CASE.B12.AB.ISsum4.sync) CASE.B12.ISs.AB.sync
```
```{bash}
cat <(echo -e "CHROM\tPOS\tREF\tISB11_RS1\tISB11_RS2\tISB11_RS3\tISB11_RS4") CASE.B11.ISs.AB.sync > CASE.B11.ISsum.AB.sync
cat <(echo -e "CHROM\tPOS\tREF\tISB10_RS1\tISB10_RS2\tISB10_RS3\tISB10_RS4") CASE.B10.ISs.AB.sync > CASE.B10.ISsum.AB.sync
cat <(echo -e "CHROM\tPOS\tREF\tISB12_RS1\tISB12_RS2\tISB12_RS3\tISB12_RS4") CASE.B12.ISs.AB.sync > CASE.B12.ISsum.AB.sync

paste <(cut -f1-4 CASE.B10.ISsum.AB.sync) <(cut -f15 CASE.All.Blocks.cov.fil.sync) <(cut -f5 CASE.B10.ISsum.AB.sync) <(cut -f19 CASE.All.Blocks.cov.fil.sync) <(cut -f6 CASE.B10.ISsum.AB.sync) <(cut -f22 CASE.All.Blocks.cov.fil.sync) <(cut -f4 CASE.B11.ISsum.AB.sync) <(cut -f16 CASE.All.Blocks.cov.fil.sync) <(cut -f5 CASE.B11.ISsum.AB.sync) <(cut -f21 CASE.All.Blocks.cov.fil.sync) <(cut -f6 CASE.B11.ISsum.AB.sync) <(cut -f23 CASE.All.Blocks.cov.fil.sync) <(cut -f7 CASE.B11.ISsum.AB.sync) <(cut -f24 CASE.All.Blocks.cov.fil.sync) <(cut -f4 CASE.B12.ISsum.AB.sync) <(cut -f17 CASE.All.Blocks.cov.fil.sync) <(cut -f5 CASE.B12.ISsum.AB.sync) <(cut -f18 CASE.All.Blocks.cov.fil.sync) <(cut -f6 CASE.B12.ISsum.AB.sync) <(cut -f20 CASE.All.Blocks.cov.fil.sync) <(cut -f7 CASE.B12.ISsum.AB.sync) <(cut -f25 CASE.All.Blocks.cov.fil.sync) > CASE.AB.sync

#paste <(cut -f1-4 CASE.B10.ISsum.AB.sync) <(cut -f15 CASE.All.Blocks.cov.fil.sync) <(cut -f5 CASE.B10.ISsum.AB.sync) <(cut -f19 CASE.All.Blocks.cov.fil.sync) <(cut -f6 CASE.B10.ISsum.AB.sync) <(cut -f22 CASE.All.Blocks.cov.fil.sync) <(cut -f4 CASE.B11.ISsum.AB.sync) <(cut -f16 CASE.All.Blocks.cov.fil.sync)  <(cut -f5 CASE.B11.ISsum.AB.sync) <(cut -f23 CASE.All.Blocks.cov.fil.sync) <(cut -f6 CASE.B11.ISsum.AB.sync) <(cut -f24 CASE.All.Blocks.cov.fil.sync) <(cut -f4 CASE.B12.ISsum.AB.sync) <(cut -f17 CASE.All.Blocks.cov.fil.sync) <(cut -f5 CASE.B12.ISsum.AB.sync) <(cut -f18 CASE.All.Blocks.cov.fil.sync) <(cut -f6 CASE.B12.ISsum.AB.sync) <(cut -f20 CASE.All.Blocks.cov.fil.sync)  > CASE.AB.sync

tail -n +2 CASE.AB.sync | mawk '$2 != 74609903 && $2 != 93711267 && $2 != 36030885' > CASE.AB.input 

paste <(cut -f1-4 CASE.B10.ISsum.AB.sync) <(cut -f7 CASE.All.Blocks.cov.fil.sync) <(cut -f5 CASE.B10.ISsum.AB.sync) <(cut -f9 CASE.All.Blocks.cov.fil.sync) <(cut -f6 CASE.B10.ISsum.AB.sync) <(cut -f12 CASE.All.Blocks.cov.fil.sync) <(cut -f4 CASE.B11.ISsum.AB.sync) <(cut -f5 CASE.All.Blocks.cov.fil.sync) <(cut -f5 CASE.B11.ISsum.AB.sync) <(cut -f8 CASE.All.Blocks.cov.fil.sync) <(cut -f6 CASE.B11.ISsum.AB.sync) <(cut -f10 CASE.All.Blocks.cov.fil.sync) <(cut -f7 CASE.B11.ISsum.AB.sync) <(cut -f14 CASE.All.Blocks.cov.fil.sync) <(cut -f4 CASE.B12.ISsum.AB.sync) <(cut -f4 CASE.All.Blocks.cov.fil.sync) <(cut -f5 CASE.B12.ISsum.AB.sync) <(cut -f6 CASE.All.Blocks.cov.fil.sync) <(cut -f6 CASE.B12.ISsum.AB.sync) <(cut -f11 CASE.All.Blocks.cov.fil.sync) <(cut -f7 CASE.B12.ISsum.AB.sync) <(cut -f13 CASE.All.Blocks.cov.fil.sync) > CA.AB.sync

#paste <(cut -f1-4 CASE.B10.ISsum.AB.sync) <(cut -f7 CASE.All.Blocks.cov.fil.sync) <(cut -f5 CASE.B10.ISsum.AB.sync) <(cut -f9 CASE.All.Blocks.cov.fil.sync) <(cut -f6 CASE.B10.ISsum.AB.sync) <(cut -f12 CASE.All.Blocks.cov.fil.sync) <(cut -f4 CASE.B11.ISsum.AB.sync) <(cut -f5 CASE.All.Blocks.cov.fil.sync)  <(cut -f5 CASE.B11.ISsum.AB.sync) <(cut -f10 CASE.All.Blocks.cov.fil.sync) <(cut -f6 CASE.B11.ISsum.AB.sync) <(cut -f14 CASE.All.Blocks.cov.fil.sync) <(cut -f4 CASE.B12.ISsum.AB.sync) <(cut -f4 CASE.All.Blocks.cov.fil.sync) <(cut -f5 CASE.B12.ISsum.AB.sync) <(cut -f6 CASE.All.Blocks.cov.fil.sync)  <(cut -f6 CASE.B12.ISsum.AB.sync) <(cut -f13 CASE.All.Blocks.cov.fil.sync) > CA.AB.sync

tail -n +2 CA.AB.sync | mawk '$2 != 93711267 && $2 != 59024150 && $2 != 2532550'> CA.AB.input 

paste <(cut -f1-4 CASE.B10.ISsum.AB.sync) <(cut -f26 CASE.All.Blocks.cov.fil.sync) <(cut -f5 CASE.B10.ISsum.AB.sync) <(cut -f30 CASE.All.Blocks.cov.fil.sync) <(cut -f6 CASE.B10.ISsum.AB.sync) <(cut -f32 CASE.All.Blocks.cov.fil.sync) <(cut -f4 CASE.B11.ISsum.AB.sync) <(cut -f27 CASE.All.Blocks.cov.fil.sync) <(cut -f5 CASE.B11.ISsum.AB.sync) <(cut -f29 CASE.All.Blocks.cov.fil.sync) <(cut -f6 CASE.B11.ISsum.AB.sync) <(cut -f33 CASE.All.Blocks.cov.fil.sync) <(cut -f7 CASE.B11.ISsum.AB.sync) <(cut -f35 CASE.All.Blocks.cov.fil.sync) <(cut -f4 CASE.B12.ISsum.AB.sync) <(cut -f28 CASE.All.Blocks.cov.fil.sync) <(cut -f5 CASE.B12.ISsum.AB.sync) <(cut -f31 CASE.All.Blocks.cov.fil.sync) <(cut -f6 CASE.B12.ISsum.AB.sync) <(cut -f34 CASE.All.Blocks.cov.fil.sync) <(cut -f7 CASE.B12.ISsum.AB.sync) <(cut -f36 CASE.All.Blocks.cov.fil.sync) > CON.AB.sync

#paste <(cut -f1-4 CASE.B10.ISsum.AB.sync) <(cut -f26 CASE.All.Blocks.cov.fil.sync) <(cut -f5 CASE.B10.ISsum.AB.sync) <(cut -f30 CASE.All.Blocks.cov.fil.sync) <(cut -f6 CASE.B10.ISsum.AB.sync) <(cut -f32 CASE.All.Blocks.cov.fil.sync) <(cut -f4 CASE.B11.ISsum.AB.sync) <(cut -f27 CASE.All.Blocks.cov.fil.sync) <(cut -f5 CASE.B11.ISsum.AB.sync) <(cut -f29 CASE.All.Blocks.cov.fil.sync) <(cut -f6 CASE.B11.ISsum.AB.sync) <(cut -f35 CASE.All.Blocks.cov.fil.sync) <(cut -f4 CASE.B12.ISsum.AB.sync) <(cut -f28 CASE.All.Blocks.cov.fil.sync) <(cut -f5 CASE.B12.ISsum.AB.sync) <(cut -f31 CASE.All.Blocks.cov.fil.sync) <(cut -f6 CASE.B12.ISsum.AB.sync) <(cut -f34 CASE.All.Blocks.cov.fil.sync)  > CON.AB.sync

tail -n +2 CON.AB.sync | mawk '$2 != 34804390 && $2 != 54895214 && $2 !=  49000140 && $2 != 15140923' > CON.AB.input 

paste <(cut -f1-4 CASE.B10.ISsum.AB.sync) <(cut -f51 CASE.All.Blocks.cov.fil.sync) <(cut -f5 CASE.B10.ISsum.AB.sync) <(cut -f52 CASE.All.Blocks.cov.fil.sync) <(cut -f6 CASE.B10.ISsum.AB.sync) <(cut -f55 CASE.All.Blocks.cov.fil.sync) <(cut -f4 CASE.B11.ISsum.AB.sync) <(cut -f49 CASE.All.Blocks.cov.fil.sync) <(cut -f5 CASE.B11.ISsum.AB.sync) <(cut -f53 CASE.All.Blocks.cov.fil.sync) <(cut -f6 CASE.B11.ISsum.AB.sync) <(cut -f54 CASE.All.Blocks.cov.fil.sync) <(cut -f7 CASE.B11.ISsum.AB.sync) <(cut -f58 CASE.All.Blocks.cov.fil.sync) <(cut -f4 CASE.B12.ISsum.AB.sync) <(cut -f50 CASE.All.Blocks.cov.fil.sync) <(cut -f5 CASE.B12.ISsum.AB.sync) <(cut -f56 CASE.All.Blocks.cov.fil.sync) <(cut -f6 CASE.B12.ISsum.AB.sync) <(cut -f57 CASE.All.Blocks.cov.fil.sync) <(cut -f7 CASE.B12.ISsum.AB.sync) <(cut -f59 CASE.All.Blocks.cov.fil.sync) > SE.AB.sync

#paste <(cut -f1-4 CASE.B10.ISsum.AB.sync) <(cut -f51 CASE.All.Blocks.cov.fil.sync) <(cut -f5 CASE.B10.ISsum.AB.sync) <(cut -f52 CASE.All.Blocks.cov.fil.sync) <(cut -f6 CASE.B10.ISsum.AB.sync) <(cut -f55 CASE.All.Blocks.cov.fil.sync) <(cut -f4 CASE.B11.ISsum.AB.sync) <(cut -f49 CASE.All.Blocks.cov.fil.sync) <(cut -f5 CASE.B11.ISsum.AB.sync) <(cut -f54 CASE.All.Blocks.cov.fil.sync) <(cut -f6 CASE.B11.ISsum.AB.sync) <(cut -f58 CASE.All.Blocks.cov.fil.sync)  <(cut -f4 CASE.B12.ISsum.AB.sync) <(cut -f56 CASE.All.Blocks.cov.fil.sync) <(cut -f5 CASE.B12.ISsum.AB.sync) <(cut -f57 CASE.All.Blocks.cov.fil.sync) <(cut -f6 CASE.B12.ISsum.AB.sync) <(cut -f59 CASE.All.Blocks.cov.fil.sync) > SE.AB.sync

tail -n +2 SE.AB.sync | mawk '$2 != 67902213 && $2 != 19828063 && $2 != 93711267 ' > SE.AB.input 

```
#### Calculate p-values
```{r}
ca.ab.sync <- read.sync(file="CA.AB.input",gen=c(0,1,0,1,0,1,0,1,0,1,0,1,0,1,0,1,0,1,0,1,0,1),repl=c(1,1,2,2,3,3,4 ,4,5,5,6,6,7,7,8,8,9,9,10,10,11,11 ))
ca.ab.cov <- coverage(ca.ab.sync,gen=c(0,1,0,1,0,1,0,1,0,1,0,1,0,1,0,1,0,1,0,1,0,1),repl=c(1,1,2,2,3, 3, 4 , 4,5,5,6,6,7,7,8,8,9,9,10,10,11,11 ))
ca.ab.af <- af(ca.ab.sync,gen=c(0, 1, 0, 1, 0, 1, 0, 1,0,1,0,1,0,1,0,1,0,1,0,1,0,1), repl=c(1, 1, 2, 2, 3, 3, 4 , 4,5,5,6,6,7,7,8,8,9,9,10,10,11,11 ))

case.ab.sync <- read.sync(file="CASE.AB.input", gen=c(0, 1, 0, 1, 0, 1, 0, 1,0,1,0,1,0,1,0,1,0,1,0,1,0,1), repl=c(1, 1, 2, 2, 3, 3, 4 , 4,5,5,6,6,7,7,8,8,9,9,10,10,11,11 ))
case.ab.cov <- coverage(case.ab.sync,gen=c(0, 1, 0, 1, 0, 1, 0, 1,0,1,0,1,0,1,0,1,0,1,0,1,0,1), repl=c(1, 1, 2, 2, 3, 3, 4 , 4,5,5,6,6,7,7,8,8,9,9,10,10,11,11 ))
case.ab.af <- af(case.ab.sync,gen=c(0, 1, 0, 1, 0, 1, 0, 1,0,1,0,1,0,1,0,1,0,1,0,1,0,1), repl=c(1, 1, 2, 2, 3, 3, 4 , 4,5,5,6,6,7,7,8,8,9,9,10,10,11,11 ))

se.ab.sync <- read.sync(file="SE.AB.input", gen=c(0, 1, 0, 1, 0, 1, 0, 1,0,1,0,1,0,1,0,1,0,1,0,1,0,1), repl=c(1, 1, 2, 2, 3, 3, 4 , 4,5,5,6,6,7,7,8,8,9,9,10,10,11,11 ))
se.ab.cov <- coverage(se.ab.sync,gen=c(0, 1, 0, 1, 0, 1, 0, 1,0,1,0,1,0,1,0,1,0,1,0,1,0,1), repl=c(1, 1, 2, 2, 3, 3, 4 , 4,5,5,6,6,7,7,8,8,9,9,10,10,11,11 ))
se.ab.af <- af(se.ab.sync,gen=c(0, 1, 0, 1, 0, 1, 0, 1,0,1,0,1,0,1,0,1,0,1,0,1,0,1), repl=c(1, 1, 2, 2, 3, 3, 4 , 4,5,5,6,6,7,7,8,8,9,9,10,10,11,11 ))

con.ab.sync <- read.sync(file="CON.AB.input", gen=c(0, 1, 0, 1, 0, 1, 0, 1,0,1,0,1,0,1,0,1,0,1,0,1,0,1), repl=c(1, 1, 2, 2, 3, 3, 4 , 4,5,5,6,6,7,7,8,8,9,9,10,10,11,11 ))
con.ab.cov <- coverage(con.ab.sync,gen=c(0, 1, 0, 1, 0, 1, 0, 1,0,1,0,1,0,1,0,1,0,1,0,1,0,1), repl=c(1, 1, 2, 2, 3, 3, 4 , 4,5,5,6,6,7,7,8,8,9,9,10,10,11,11 ))
con.ab.af <- af(con.ab.sync,gen=c(0, 1, 0, 1, 0, 1, 0, 1,0,1,0,1,0,1,0,1,0,1,0,1,0,1), repl=c(1, 1, 2, 2, 3, 3, 4 , 4,5,5,6,6,7,7,8,8,9,9,10,10,11,11 ))
```

```{r}
ca.ab.pvals <- adapted.cmh.test(freq=ca.ab.af, coverage=ca.ab.cov, Ne=c(250,250,250,1000,1000,1000,1000,1000,1000,1000,1000), gen=c(0,1,0,1,0,1,0,1,0,1,0,1,0,1,0,1,0,1,0,1,0,1),repl=c(1,1,2,2,3,3,4 ,4,5,5,6,6,7,7,8,8,9,9,10,10,11,11 ), poolSize=rep(c(40000,10000), ncol(ca.ab.af)/2), MeanStart = TRUE, mincov =14)

pcadf <- as.data.table(cbind(row.names(ca.ab.af),ca.ab.pvals))
colnames(pcadf) <- c("Locus","PVAL")
pcadf$GROUP <- "PCA"

case.ab.pvals <- adapted.cmh.test(freq=case.ab.af, coverage=case.ab.cov, Ne=c(250,250,250,1000,1000,1000,1000,1000,1000,1000,1000), gen=c(0,1,0,1,0,1,0,1,0,1,0,1,0,1,0,1,0,1,0,1,0,1),repl=c(1,1,2,2,3,3,4 ,4,5,5,6,6,7,7,8,8,9,9,10,10,11,11 ), poolSize=rep(c(40000,10000), ncol(case.ab.af)/2), MeanStart = TRUE, mincov =14)

pcasedf <- as.data.table(cbind(row.names(case.ab.af),case.ab.pvals))
colnames(pcasedf) <- c("Locus","PVAL")
pcasedf$GROUP <- "PCASE"

se.ab.pvals <- adapted.cmh.test(freq=se.ab.af, coverage=se.ab.cov, Ne=c(250,250,250,1000,1000,1000,1000,1000,1000,1000,1000), gen=c(0,1,0,1,0,1,0,1,0,1,0,1,0,1,0,1,0,1,0,1,0,1),repl=c(1,1,2,2,3,3,4 ,4,5,5,6,6,7,7,8,8,9,9,10,10,11,11 ), poolSize=rep(c(40000,10000), ncol(se.ab.af)/2), MeanStart = TRUE, mincov =14)

psedf <- as.data.table(cbind(row.names(se.ab.af),se.ab.pvals))
colnames(psedf) <- c("Locus","PVAL")
psedf$GROUP <- "PSE"

con.ab.pvals <- adapted.cmh.test(freq=con.ab.af, coverage=con.ab.cov, Ne=c(250,250,250,1000,1000,1000,1000,1000,1000,1000,1000), gen=c(0,1,0,1,0,1,0,1,0,1,0,1,0,1,0,1,0,1,0,1,0,1),repl=c(1,1,2,2,3,3,4 ,4,5,5,6,6,7,7,8,8,9,9,10,10,11,11 ), poolSize=rep(c(40000,10000), ncol(con.ab.af)/2), MeanStart = TRUE, mincov =14)

pcondf <- as.data.table(cbind(row.names(con.ab.af),con.ab.pvals)) 
colnames(pcondf) <- c("Locus","PVAL")
pcondf$GROUP <- "PCON"


AB.pval.table <- bind_rows(pcadf,pcasedf,pcondf,psedf)
AB.pval.table <- AB.pval.table %>%
  separate(Locus, into = c("CHROM", "BP"), sep = "\\.(?=[^.]+$)")
AB.pval.table$PVAL <- as.numeric(AB.pval.table$PVAL)
```


```{r}
Qvalue_convert(AB.pval.table) -> AB.pv
```
```{r}

alpha = 0.01
alpha2 =0.1

ppsig.AB <- Significat_subset(AB.pv,alpha,alpha2)


all_sig.AB.CA<- subset(ppsig.AB, Sig.CA == "TRUE") 
all_sig.AB.CA$Sig.CASE <- FALSE
all_sig.AB.CA$Sig.SE <- FALSE

all_sig.AB.CASE<- subset(ppsig.AB, Sig.CASE == "TRUE") 
all_sig.AB.CASE$Sig.SE <- FALSE
all_sig.AB.CASE$Sig.CA <- FALSE

all_sig.AB.SE<- subset(ppsig.AB, Sig.SE == "TRUE") 
all_sig.AB.SE$Sig.CASE <- FALSE
all_sig.AB.SE$Sig.CA <- FALSE

all_sig.AB.AT <- rbind(all_sig.AB.CA,all_sig.AB.CASE,all_sig.AB.SE)

write.table(all_sig.AB.AT, "Sig.Loci.FDR01.AB", sep="\t", row.names = FALSE, quote = FALSE)


alphaAB = 0.0001
alpha2 =0.1


ppsig1.ab <-Significat_subset(AB.pv,alphaAB,alpha2)
write.table(ppsig1.ab, "Sig.Loci.FDR0001.AB", sep="\t", row.names = FALSE, quote = FALSE)

conpp.AB <- subset(AB.pv, QCON < 0.001)
all_con <- bind_rows(conpp1,conpp.AB) %>% group_by(SNP) %>% filter(n()>1)
#all_con <- bind_rows(conpp1,conpp.AB) 
all_con <- subset(all_con,QCON / QCASE < 10 & QCON / QCA < 10 & QCON/ QSE < 10)

write.table(all_con, "ALL.CON.SNPS", sep="\t", row.names = FALSE, quote = FALSE)

ppsig.AB$BLOCK <- 33
ppsig1.ab$BLOCK <-33

#multiple.sig.o<- bind_rows(ppsig1 %>% distinct(SNP, .keep_all = TRUE),ppsig1.ab) %>% group_by(SNP) %>% filter(n()>1)

multiple.sig.SE <- bind_rows(subset(ppsig3.FDR01, Sig.SE == TRUE), subset(ppsig1.ab, Sig.SE == TRUE)) %>% group_by(SNP) %>% filter(n()>1)
multiple.sig.SE$Sig.CA <- FALSE
multiple.sig.SE$Sig.CASE <- FALSE

multiple.sig.CASE <- bind_rows(subset(ppsig3.FDR01, Sig.CASE == TRUE), subset(ppsig1.ab, Sig.CASE == TRUE)) %>% group_by(SNP) %>% filter(n()>1)
multiple.sig.CASE$Sig.CA <- FALSE
multiple.sig.CASE$Sig.SE <- FALSE

multiple.sig.CA<- bind_rows(subset(ppsig3.FDR01, Sig.CA == TRUE), subset(ppsig1.ab, Sig.CA == TRUE)) %>% group_by(SNP) %>% filter(n()>1)
multiple.sig.CA$Sig.SE <- FALSE
multiple.sig.CA$Sig.CASE <- FALSE

multiple.sig.o <- bind_rows(multiple.sig.SE, multiple.sig.CASE, multiple.sig.CA)

total.sig <- anti_join(bind_rows(multiple.sig.o,all_sig,multi_sig, multi_sig.01, ppsig1.s),all_con, by = "SNP")

write.table(total.sig, "Total.Significant.uf.Loci", sep="\t", row.names = FALSE, quote = FALSE)
```

```{bash}
source activate CASE

cut -f2,3 Total.Significant.uf.Loci | tail -n +2 | sort -k1,2 | uniq | mawk '{print $1 "\t" $2-1 "\t" $2}' | bedtools sort -i - > CASE.Total.Significant.uf.loci.bed

bedtools intersect -wb -a  <(mawk '{print $2"\t"$3-1"\t"$3}' ALL.CON.SNPS ) -b ~/CASE/analysis/sorted.ref3.0.gene.bed | grep -oh "gene=LOC.*;g" | sed 's/gene=//g' | sed 's/;g//g' | sort | uniq > CON.LOCI

bedtools intersect -wb -a  <(mawk '{print $2"\t"$3-1"\t"$3}' ALL.CON.SNPS ) -b ~/CASE/analysis/sorted.ref3.0.gene.bed | cut -f4- | bedtools sort | bedtools merge > con.loci.filter.bed

cat <(echo -e "SNP\tCHROM\tBP") <(bedtools intersect -a CASE.Total.Significant.uf.loci.bed -b con.loci.filter.bed | mawk '{print $1"_"$3"\t"$1"\t"$3}') > ToFilter.Con.Loci.txt
```


```{r}
con_filter <- read.table("ToFilter.Con.Loci.txt",header=TRUE)

total.sig <- unique(anti_join(total.sig,con_filter, by = "SNP"))

write.table(total.sig, "Total.Significant.Loci", sep="\t", row.names = FALSE, quote = FALSE)


grouped_total.sig <- group_and_average(total.sig)

write.table(grouped_total.sig, "Total.Significant.Grouped.Loci", sep="\t", row.names = FALSE, quote = FALSE)

totalsig.CA <- subset(total.sig, Sig.CA == TRUE & Sig.SE == FALSE )
totalsig.SE <- subset(total.sig, Sig.CA == FALSE & Sig.SE == TRUE )
totalsig.both <- subset(total.sig, Sig.CA == TRUE & Sig.SE == TRUE )
totalsig.CASE <- subset(total.sig, Sig.CASE == TRUE )

write.table(totalsig.CA, "sig.ca", sep="\t", row.names = FALSE, quote = FALSE)
write.table(totalsig.SE, "sig.se", sep="\t", row.names = FALSE, quote = FALSE)
write.table(totalsig.both, "sig.both", sep="\t", row.names = FALSE, quote = FALSE)
write.table(totalsig.CASE, "sig.case", sep="\t", row.names = FALSE, quote = FALSE)

```

```{bash}
cat <(echo -e "SNP\tGroup") <(mawk '{if (NR > 1) print $1"\tSE"}' sig.se ) <(mawk '{if (NR > 1) print $1"\tCA"}' sig.ca ) <(mawk '{if (NR > 1) print $1"\tBoth"}' sig.both ) > sig.table
```


```{bash}
source activate CASE

cut -f2,3 Total.Significant.Loci | tail -n +2 | sort -k1,2 | uniq | mawk '{print $1 "\t" $2-1 "\t" $2}' | bedtools sort -i - > CASE.Total.Significant.loci.bed
bedtools slop -b 100 -i CASE.Total.Significant.loci.bed  -g ../reference.fasta.fai  | bedtools merge -i - >  CASE.Total.Significant.loci.intervals.bed


bedtools intersect -wb -a CASE.Total.Significant.loci.bed -b ~/CASE/analysis/sorted.ref3.0.gene.bed | grep -oh "gene=LOC.*;g" | sed 's/gene=//g' | sed 's/;g//g' | sort | uniq > Sig.loci.TOTAL.CASE.LOC

``` 


```{bash}
source activate CASE

mawk '$14 == "TRUE" ' Total.Significant.Loci | sort -k1,2 | mawk '{print $2 "\t" $3-1 "\t" $3}' | uniq  > CA.Significant.loci.bed
bedtools intersect -wb -a CA.Significant.loci.bed -b ~/CASE/analysis/sorted.ref3.0.gene.bed | grep -oh "gene=LOC.*;g" | sed 's/gene=//g' | sed 's/;g//g' | sort | uniq > Sig.loci.1.CA.LOC

mawk '$15 =="TRUE"' Total.Significant.Loci | sort -k1,2  |  mawk '{print $2 "\t" $3-1 "\t" $3}'| uniq  > SE.Significant.loci.bed
bedtools intersect -wb -a SE.Significant.loci.bed -b ~/CASE/analysis/sorted.ref3.0.gene.bed | grep -oh "gene=LOC.*;g" | sed 's/gene=//g' | sed 's/;g//g' | sort | uniq > Sig.loci.1.SE.LOC

mawk '$13 == "TRUE"' Total.Significant.Loci | sort -k1,2 |   mawk '{print $2 "\t" $3-1 "\t" $3}' | uniq  > CASE.Significant.loci.bed
bedtools intersect -wb -a CASE.Significant.loci.bed -b ~/CASE/analysis/sorted.ref3.0.gene.bed | grep -oh "gene=LOC.*;g" | sed 's/gene=//g' | sed 's/;g//g' | sort | uniq > Sig.loci.1.CASE.LOC

mawk '$11 == "TRUE" && $13 == "TRUE"  && $12 == "FALSE"' Total.Significant.Grouped.Loci | mawk '{print $14 "\t" $15-1 "\t" $15}'| uniq  > SE.CASE.Significant.loci.bed

bedtools intersect -wb -a SE.CASE.Significant.loci.bed -b ~/CASE/analysis/sorted.ref3.0.gene.bed | grep -oh "gene=LOC.*;g" | sed 's/gene=//g' | sed 's/;g//g' | sort | uniq > Sig.loci.SE.CASE.LOC

mawk '$11 == "TRUE" && $13 == "TRUE"  && $12 == "TRUE"' Total.Significant.Grouped.Loci | mawk '{print $14 "\t" $15-1 "\t" $15}'| uniq  > SE.CASE.CA.Significant.loci.bed

bedtools intersect -wb -a SE.CASE.CA.Significant.loci.bed -b ~/CASE/analysis/sorted.ref3.0.gene.bed | grep -oh "gene=LOC.*;g" | sed 's/gene=//g' | sed 's/;g//g' | sort | uniq > Sig.loci.SE.CASE.CA.LOC

mawk '$11 == "FALSE" && $13 == "TRUE"  && $12 == "TRUE"' Total.Significant.Grouped.Loci | mawk '{print $14 "\t" $15-1 "\t" $15}'| uniq  > SE.CA.Significant.loci.bed

mawk '$11 == "TRUE" && $13 == "FALSE"  && $12 == "TRUE"' Total.Significant.Grouped.Loci | mawk '{print $14 "\t" $15-1 "\t" $15}'| uniq  > CA.CASE.Significant.loci.bed

mawk '$11 == "FALSE" && $13 == "TRUE"  && $12 == "FALSE"' Total.Significant.Grouped.Loci | mawk '{print $14 "\t" $15-1 "\t" $15}'| uniq  > SE.ONLY.Significant.loci.bed

mawk '$11 == "TRUE" && $13 == "FALSE"  && $12 == "FALSE"' Total.Significant.Grouped.Loci | mawk '{print $14 "\t" $15-1 "\t" $15}'| uniq  > CASE.ONLY.Significant.loci.bed

mawk '$11 == "FALSE" && $13 == "FALSE"  && $12 == "TRUE"' Total.Significant.Grouped.Loci | mawk '{print $14 "\t" $15-1 "\t" $15}'| uniq  > CA.ONLY.Significant.loci.bed

cat CA.ONLY.Significant.loci.bed CA.CASE.Significant.loci.bed| sort|  uniq  > CA.factor.Significant.loci.bed

cat SE.ONLY.Significant.loci.bed SE.CASE.Significant.loci.bed| sort | uniq  > SE.factor.Significant.loci.bed
```

```{bash}
source activate CASE


#B10
mawk '$14 == "TRUE"' Sig.Loci.FDR01.Block10 | sort -k1,2  | mawk '{print $2 "\t" $3-1 "\t" $3}'| uniq  > CA.Significant.B10.loci.bed
bedtools intersect -wb -a CA.Significant.B10.loci.bed -b ~/CASE/analysis/sorted.ref3.0.gene.bed | grep -oh "gene=LOC.*;g" | sed 's/gene=//g' | sed 's/;g//g' | sort | uniq > Sig.loci.B10.CA.LOC

mawk '$15 == "TRUE" ' Sig.Loci.FDR01.Block10 | sort -k1,2  |  mawk '{print $2 "\t" $3-1 "\t" $3}' | uniq  > SE.Significant.B10.loci.bed
bedtools intersect -wb -a SE.Significant.B10.loci.bed -b ~/CASE/analysis/sorted.ref3.0.gene.bed | grep -oh "gene=LOC.*;g" | sed 's/gene=//g' | sed 's/;g//g' | sort | uniq > Sig.loci.B10.SE.LOC

mawk '$13 == "TRUE" ' Sig.Loci.FDR01.Block10 | sort -k1,2 |  mawk '{print $2 "\t" $3-1 "\t" $3}' | uniq  > CASE.Significant.B10.loci.bed
bedtools intersect -wb -a CASE.Significant.B10.loci.bed -b ~/CASE/analysis/sorted.ref3.0.gene.bed | grep -oh "gene=LOC.*;g" | sed 's/gene=//g' | sed 's/;g//g' | sort | uniq > Sig.loci.B10.CASE.LOC

#B11
mawk '$14 == "TRUE"' Sig.Loci.FDR01.Block11 | sort -k1,2  | mawk '{print $2 "\t" $3-1 "\t" $3}'  | uniq > CA.Significant.B11.loci.bed
bedtools intersect -wb -a CA.Significant.B11.loci.bed -b ~/CASE/analysis/sorted.ref3.0.gene.bed | grep -oh "gene=LOC.*;g" | sed 's/gene=//g' | sed 's/;g//g' | sort | uniq > Sig.loci.B11.CA.LOC

mawk '$15 == "TRUE" ' Sig.Loci.FDR01.Block11 | sort -k1,2 |  mawk '{print $2 "\t" $3-1 "\t" $3}' | uniq > SE.Significant.B11.loci.bed
bedtools intersect -wb -a SE.Significant.B11.loci.bed -b ~/CASE/analysis/sorted.ref3.0.gene.bed | grep -oh "gene=LOC.*;g" | sed 's/gene=//g' | sed 's/;g//g' | sort | uniq > Sig.loci.B11.SE.LOC

mawk '$13 == "TRUE" ' Sig.Loci.FDR01.Block11 | sort -k1,2  |  mawk '{print $2 "\t" $3-1 "\t" $3}' | uniq  > CASE.Significant.B11.loci.bed
bedtools intersect -wb -a CASE.Significant.B11.loci.bed -b ~/CASE/analysis/sorted.ref3.0.gene.bed | grep -oh "gene=LOC.*;g" | sed 's/gene=//g' | sed 's/;g//g' | sort | uniq > Sig.loci.B11.CASE.LOC

#B12
mawk '$14 == "TRUE" ' Sig.Loci.FDR01.Block12 | sort -k1,2 | mawk '{print $2 "\t" $3-1 "\t" $3}' | uniq  > CA.Significant.B12.loci.bed
bedtools intersect -wb -a CA.Significant.B12.loci.bed -b ~/CASE/analysis/sorted.ref3.0.gene.bed | grep -oh "gene=LOC.*;g" | sed 's/gene=//g' | sed 's/;g//g' | sort | uniq > Sig.loci.B12.CA.LOC

mawk '$15 == "TRUE" ' Sig.Loci.FDR01.Block12 | sort -k1,2  |  mawk '{print $2 "\t" $3-1 "\t" $3}' | uniq  > SE.Significant.B12.loci.bed
bedtools intersect -wb -a SE.Significant.B12.loci.bed -b ~/CASE/analysis/sorted.ref3.0.gene.bed | grep -oh "gene=LOC.*;g" | sed 's/gene=//g' | sed 's/;g//g' | sort | uniq > Sig.loci.B12.SE.LOC

mawk '$13 == "TRUE" ' Sig.Loci.FDR01.Block12 | sort -k1,2  |  mawk '{print $2 "\t" $3-1 "\t" $3}' | uniq  > CASE.Significant.B12.loci.bed
bedtools intersect -wb -a CASE.Significant.B12.loci.bed -b ~/CASE/analysis/sorted.ref3.0.gene.bed | grep -oh "gene=LOC.*;g" | sed 's/gene=//g' | sed 's/;g//g' | sort | uniq > Sig.loci.B12.CASE.LOC
``` 


```{r}
source("../gprofiler2W.R")
#library(gprofiler2)
#set_base_url("http://biit.cs.ut.ee/gprofiler_beta")


gostres <- gost(query = list("CASE" = read.table("Sig.loci.1.CASE.LOC",header=FALSE), "SE" = read.table("Sig.loci.1.SE.LOC",header=FALSE), "CA" = read.table("Sig.loci.1.CA.LOC",header=FALSE)), organism = "cvgca002022765v4", multi_query = FALSE,user_threshold = 0.05,domain_scope = "known")

p <-gostplot(gostres, capped = TRUE, interactive = FALSE)

p <- p+ theme_black() + theme(legend.position = "none")

pp <- publish_gostplot(p, width = NA, height = NA, filename = NULL )

png(filename="TestGO.png", type="cairo",units="px", width=4000, height=2500, res=300, bg="transparent")

pp 
dev.off()
```



# Manhattan Plots
#### CASE
```{r, warning =FALSE, eval = FALSE}
b10.h <- subset(total.sig,  BLOCK == "10" & Sig.CASE == TRUE)
b10.c <- subset(B10.pv, QCON < 0.1)


B10.pp <- B10.pv %>%
  mutate(
    Group = case_when(
      SNP %in% singleton$SNP ~ "Singleton",
      SNP %in% b10.c$SNP ~ "Control Treatment Filtered",
      SNP %in% all_con$SNP ~ "Control Treatment Filtered",
      SNP %in% con_filter$SNP ~ "Control Treatment Filtered",
      TRUE ~ "Shared SNP"
    )
  )

man <-ggman(B10.pp, pvalue="QCASE", chr="CHR", snp="SNP",bp="BP", relative.positions = TRUE ,sigLine = NA, pointSize=1.5,title = "", alpha = 0.5)

dfm <- man[[1]]

dfm$chrom_alt <- factor(dfm$chrom_alt, levels=c(0,1))

dfmo <- semi_join(dfm, b10.h, by = "SNP")
dfmo$Group <- "Outlier"

dfmsplit <- split(dfm, dfm$chrom)
xbreaks <- sapply(dfmsplit,function(x) {
    midpoint <- length(x$index)/2
    if(midpoint <1) midpoint <- 1
    return(x$index[midpoint])
})
b10.CASE <-ggplot(dfm, aes(x= index, y=marker, colour = as.factor(chrom_alt)))+
  geom_point(aes(alpha=marker, shape=Group, size = marker))+  
  geom_point(data=dfmo, fill=cbPaletteSmall4[4], aes(x=index, y=marker, shape=Group, size=marker),  color = cbPaletteSmall4[4])+
  scale_x_continuous(breaks = xbreaks, labels = names(xbreaks),expand = c(0,0),limits = c(0,max(dfm$index)+10)) +
  guides(colour = FALSE,alpha=FALSE, fill = FALSE, size = FALSE) +
  scale_y_continuous(expand = c(0,0),limits=c(0,max(dfm$marker+1,na.rm=TRUE)))+
  scale_size_continuous(range = c(0.25,3)) +
  scale_alpha_continuous(range=c(0.05,0.8), limits= c(0,50))+
  ggtitle("Block 10") +
  labs(x = "Chromosome") +
  theme_classic()+
  labs(y=expression(-Log10(q-value))) +
  scale_shape_manual(values=c(15,21,16,17))+
  scale_color_manual(values=c("grey", "dark grey"), guide =FALSE)+
  guides(shape = guide_legend(override.aes = list(color = "grey", size = 5, linetype =0) ) )

b11.h <- subset(total.sig,  BLOCK == "11" & Sig.CASE == TRUE)
b11.c <- subset(B11.pv, QCON < 0.1)

B11.pp <- B11.pv %>%
  mutate(
    Group = case_when(
      SNP %in% singleton$SNP ~ "Singleton",
      SNP %in% b11.c$SNP ~ "Control Treatment Filtered",
      SNP %in% all_con$SNP ~ "Control Treatment Filtered",
      SNP %in% con_filter$SNP ~ "Control Treatment Filtered",
      TRUE ~ "Shared SNP"
    )
  )

man <-ggman(B11.pp, pvalue="QCASE", chr="CHR", snp="SNP",bp="BP", relative.positions = TRUE ,sigLine = NA, pointSize=1.5,title = "", alpha = 0.5)

dfm <- man[[1]]

dfm$chrom_alt <- factor(dfm$chrom_alt, levels=c(0,1))


dfmo <- semi_join(dfm, b11.h, by = "SNP")
dfmo$Group <- "Outlier"

dfmsplit <- split(dfm, dfm$chrom)
xbreaks <- sapply(dfmsplit,function(x) {
    midpoint <- length(x$index)/2
    if(midpoint <1) midpoint <- 1
    return(x$index[midpoint])
})

b11.CASE <-ggplot(dfm, aes(x= index, y=marker, colour = as.factor(chrom_alt)))+
  geom_point(aes(alpha=marker, shape=Group, size = marker))+  
  #geom_point(aes(x=index, y=log10(QCON), colour = as.factor(chrom_alt), alpha=0.5, shape = Group))+  
  geom_point(data=dfmo, fill=cbPaletteSmall4[4], aes(x=index, y=marker, shape=Group, size=marker),  color = cbPaletteSmall4[4])+
  scale_x_continuous(breaks = xbreaks, labels = names(xbreaks),expand = c(0,0),limits = c(0,max(dfm$index)+10)) +
  guides(colour = FALSE,alpha=FALSE, fill = FALSE, size=FALSE) +
  scale_y_continuous(expand = c(0,0),limits=c(0,max(dfm$marker+1,na.rm=TRUE)))+
  scale_size_continuous(range = c(0.25,3)) +
  scale_alpha_continuous(range=c(0.05,0.8), limits= c(0,8))+
  labs(x = "Chromosome") +
  ggtitle("Block 11") +
  theme_classic()+
  labs(y=expression(-Log10(q-value))) +
  scale_shape_manual(values=c(15,21,16,17))+
  scale_color_manual(values=c("grey", "dark grey"), guide =FALSE) +    
  guides(shape = guide_legend(override.aes = list(color = "grey", size = 5, linetype =0) ) )


b12.h <- subset(total.sig,  BLOCK == "12" & Sig.CASE == TRUE)
b12.c <- subset(B12.pv, QCON < 0.1)

B12.pp <- B12.pv %>%
  mutate(
    Group = case_when(
      SNP %in% singleton$SNP ~ "Singleton",
      SNP %in% b12.c$SNP ~ "Control Treatment Filtered",
      SNP %in% all_con$SNP ~ "Control Treatment Filtered",
      SNP %in% con_filter$SNP ~ "Control Treatment Filtered",
      TRUE ~ "Shared SNP"
    )
  )

man <-ggman(B12.pp, pvalue="QCASE", chr="CHR", snp="SNP",bp="BP", relative.positions = TRUE ,sigLine = NA, pointSize=1.5,title = "", alpha = 0.5)

dfm <- man[[1]]

dfm$chrom_alt <- factor(dfm$chrom_alt, levels=c(0,1))

dfmo <- semi_join(dfm, b12.h, by = "SNP")
dfmo$Group <- "Outlier"

dfmsplit <- split(dfm, dfm$chrom)
xbreaks <- sapply(dfmsplit,function(x) {
    midpoint <- length(x$index)/2
    if(midpoint <1) midpoint <- 1
    return(x$index[midpoint])
})

b12.CASE <-ggplot(dfm, aes(x= index, y=marker, colour = as.factor(chrom_alt)))+
  geom_point(aes(alpha=marker, shape=Group, size = marker))+  
  geom_point(data=dfmo, fill=cbPaletteSmall4[4], aes(x=index, y=marker, shape=Group, size=marker),  color = cbPaletteSmall4[4])+
  scale_x_continuous(breaks = xbreaks, labels = names(xbreaks),expand = c(0,0),limits = c(0,max(dfm$index)+10)) +
  guides(colour = FALSE,alpha=FALSE, fill = FALSE, size=FALSE) +
  #scale_y_continuous(expand = c(0,0),limits=c(0,max(dfm$marker+1,na.rm=TRUE)))+
  scale_y_continuous(expand = c(0,0),limits=c(0,20))+ #Removes two outliers near end of CHR5
  labs(x = "Chromosome") +
  ggtitle("Block 12") +
  scale_alpha_continuous(range=c(0.05,0.8), limits= c(0,20))+
  scale_size_continuous(range = c(0.25,3), limits=c(0,20)) +
  theme_classic()+
  labs(y=expression(-Log10(q-value))) +
  scale_shape_manual(values=c(15,21,16,17))+
  scale_color_manual(values=c("grey", "dark grey"), guide =FALSE) + 
  guides(shape = guide_legend(override.aes = list(color = "grey", size = 5, linetype =0) ) )
```
```{r}
png(filename="CASEmpHighlighted.png", type="cairo",units="px", width=5600, height=3000, res=300, bg="transparent")

(b10.CASE / b11.CASE /b12.CASE )  + plot_annotation(tag_levels = 'A') + plot_layout(guides = 'collect') 

null <- dev.off()
```

```{r}
png(filename="CASEmpHighlightedBB.png", type="cairo",units="px", width=5600, height=3000, res=300, bg="transparent")



(b10.CASE / b11.CASE /b12.CASE )  + plot_annotation(tag_levels = 'A') + plot_layout(guides = 'collect') & theme_black()
#(b10.CASE / b11.CASE /b12.CASE )  + plot_annotation(tag_levels = 'A') & theme_black()
#man+scale_fill_manual(values = c("#009E73","#E69F00", "#0072B2"),name = "Legend")
null <- dev.off()
```

```{r}
png(filename="CASEmpHighlightedBB12.png", type="cairo",units="px", width=5600, height=3000, res=300, bg="transparent")

b12.CASEbb <-ggplot(dfm, aes(x= index, y=marker, colour = as.factor(chrom_alt)))+
  geom_point(aes(alpha=marker, shape=Group, size = marker, show.legend = FALSE))+  
  geom_point(data=dfmo, fill=cbPaletteSmall4[4], aes(x=index, y=marker, shape=Group, size=8),  color = cbPaletteSmall4[4], alpha=0.9)+
  scale_alpha_continuous(range=c(0.05,0.8), limits= c(0,20))+
  scale_x_continuous(breaks = xbreaks, labels = names(xbreaks),expand = c(0,0),limits = c(0,max(dfm$index)+10)) +
  #scale_y_continuous(expand = c(0,0),limits=c(0,max(dfm$marker+1,na.rm=TRUE)))+
  scale_y_continuous(expand = c(0,0),limits=c(0,15))+ #Removes two singleton loci at end of CHR 5
  labs(x = "Chromosome") +
  theme_black()+
  theme(legend.text = element_text(size = 18), legend.title = element_text(size=24), legend.position = "bottom")+ 
  scale_size_continuous(range = c(0.5,5), limits= c(0,15)) +
  ggtitle("Block 12") +
  labs(y=expression(-Log10(q-value))) +
  scale_shape_manual(values=c(15,21,16,17))+
  scale_color_manual(values=c("grey", "dark grey"))+
  guides(colour = FALSE,alpha=FALSE, fill = FALSE, size=FALSE) +
  guides(shape = guide_legend(override.aes = list(color = "grey", size = 6, linetype =0) ) )
#guides(color = guide_legend(override.aes = list(color = c("grey", "grey"))))
b12.CASEbb
null <- dev.off()
```

```{r}
ball.h <- subset(total.sig, Sig.CASE == TRUE)
ball.h <- ball.h %>%
  group_by(SNP) %>%
  slice_min(QCASE)

man <-ggman(ball.h, pvalue="QCASE", chr="CHR", snp="SNP",bp="BP", relative.positions = TRUE ,sigLine = NA, pointSize=1.5,title = "", alpha = 0.5)

dfm <- man[[1]]

dfm$chrom_alt <- factor(dfm$chrom_alt, levels=c(0,1))

dfmo <- semi_join(dfm, b12.h, by = "SNP")
dfmo$Group <- "Outlier"

dfmsplit <- split(dfm, dfm$chrom)
xbreaks <- sapply(dfmsplit,function(x) {
    midpoint <- length(x$index)/2
    if(midpoint <1) midpoint <- 1
    return(x$index[midpoint])
})

ball.CASE <-ggplot(dfm, aes(x= index, y=marker, colour = as.factor(chrom_alt)))+
  geom_point(data=dfm, fill=cbPaletteSmall4[4], aes(x=index, y=marker, size=marker),  color = cbPaletteSmall4[4])+
  scale_x_continuous(breaks = xbreaks, labels = names(xbreaks),expand = c(0,0),limits = c(0,max(dfm$index)+10)) +
  guides(colour = FALSE,alpha=FALSE, fill = FALSE, size=FALSE) +
  scale_y_continuous(expand = c(0,0),limits=c(0,max(dfm$marker+1,na.rm=TRUE)))+
  #scale_y_continuous(expand = c(0,0),limits=c(0,20))+ #Removes two outliers near end of CHR5
  labs(x = "Chromosome") +
  scale_alpha_continuous(range=c(0.05,0.8), limits= c(0,20))+
  scale_size_continuous(range = c(0.25,3), limits=c(0,20)) +
  theme_classic()+
  labs(y=expression(-Log10(q-value))) +
  scale_color_manual(values=c("grey", "dark grey"), guide =FALSE) 

png(filename="CASEmpBB.png", type="cairo",units="px", width=4800, height=1000, res=300, bg="transparent")  
ball.CASE + theme_black()
null <- dev.off()

```

#### SE
```{r, warning =FALSE, eval = FALSE}
b10.h <- subset(total.sig,  BLOCK == "10" & Sig.SE == TRUE)
b10.c <- subset(B10.pv, QCON < 0.1)

B10.pp <- B10.pv %>%
  mutate(
    Group = case_when(
      SNP %in% singleton$SNP ~ "Singleton",
      SNP %in% b10.c$SNP ~ "Control Treatment Filtered",
      SNP %in% all_con$SNP ~ "Control Treatment Filtered",
      SNP %in% con_filter$SNP ~ "Control Treatment Filtered",
      TRUE ~ "Shared SNP"
    )
  )

man <-ggman(B10.pp, pvalue="QSE", chr="CHR", snp="SNP",bp="BP", relative.positions = TRUE ,sigLine = NA, pointSize=1.5,title = "", alpha = 0.5)

dfm <- man[[1]]

dfm$chrom_alt <- factor(dfm$chrom_alt, levels=c(0,1))

dfmo <- semi_join(dfm, b10.h, by = "SNP")
dfmo$Group <- "Outlier"

dfmsplit <- split(dfm, dfm$chrom)
xbreaks <- sapply(dfmsplit,function(x) {
    midpoint <- length(x$index)/2
    if(midpoint <1) midpoint <- 1
    return(x$index[midpoint])
})

b10.SE <-ggplot(dfm, aes(x= index, y=marker, colour = as.factor(chrom_alt)))+
  geom_point(aes(alpha=marker, shape=Group, size = marker))+  
  geom_point(data=dfmo, fill=cbPaletteSmall4[3], aes(x=index, y=marker, shape=Group, size=marker),  color = cbPaletteSmall4[3])+
  scale_x_continuous(breaks = xbreaks, labels = names(xbreaks),expand = c(0,0),limits = c(0,max(dfm$index)+10)) +
  guides(colour = FALSE,alpha=FALSE, fill = FALSE, size=FALSE) +
  scale_y_continuous(expand = c(0,0),limits=c(0,max(dfm$marker+1,na.rm=TRUE)))+
  ggtitle("Block 10") +
  labs(x = "Chromosome") +
  scale_size_continuous(range = c(0.25,3)) +
  scale_alpha_continuous(range=c(0.05,0.8), limits= c(0,50))+
  theme_classic()+
  labs(y=expression(-Log10(q-value))) +
  scale_shape_manual(values=c(15,21,16,17))+
  scale_color_manual(values=c("grey", "dark grey"), guide =FALSE) + 
  guides(shape = guide_legend(override.aes = list(color = "grey", size = 5, linetype =0) ) )

b11.h <- subset(total.sig,  BLOCK == "11" & Sig.SE == TRUE)
b11.c <- subset(B11.pv, QCON < 0.1)

B11.pp <- B11.pv %>%
  mutate(
    Group = case_when(
      SNP %in% singleton$SNP ~ "Singleton",
      SNP %in% b11.c$SNP ~ "Control Treatment Filtered",
      SNP %in% all_con$SNP ~ "Control Treatment Filtered",
      SNP %in% con_filter$SNP ~ "Control Treatment Filtered",
      TRUE ~ "Shared SNP"
    )
  )

man <-ggman(B11.pp, pvalue="QSE", chr="CHR", snp="SNP",bp="BP", relative.positions = TRUE ,sigLine = NA, pointSize=1.5,title = "", alpha = 0.5)

dfm <- man[[1]]

dfm$chrom_alt <- factor(dfm$chrom_alt, levels=c(0,1))


dfmo <- semi_join(dfm, b11.h, by = "SNP")
dfmo$Group <- "Outlier"

dfmsplit <- split(dfm, dfm$chrom)
xbreaks <- sapply(dfmsplit,function(x) {
    midpoint <- length(x$index)/2
    if(midpoint <1) midpoint <- 1
    return(x$index[midpoint])
})

b11.SE <-ggplot(dfm, aes(x= index, y=marker, colour = as.factor(chrom_alt)))+
  geom_point(aes(alpha=marker, shape=Group, size = marker))+  
  #geom_point(aes(x=index, y=log10(QCON), colour = as.factor(chrom_alt), alpha=0.5, shape = Group))+  
  geom_point(data=dfmo, fill=cbPaletteSmall4[3], aes(x=index, y=marker, shape=Group, size=marker),  color = cbPaletteSmall4[3])+
  scale_x_continuous(breaks = xbreaks, labels = names(xbreaks),expand = c(0,0),limits = c(0,max(dfm$index)+10)) +
  guides(colour = FALSE,alpha=FALSE, fill = FALSE, size=FALSE) +
  scale_y_continuous(expand = c(0,0),limits=c(0,max(dfm$marker+1,na.rm=TRUE)))+
  labs(x = "Chromosome") +
  ggtitle("Block 11") +
  theme_classic()+
  scale_size_continuous(range = c(0.25,3)) +
  labs(y=expression(-Log10(q-value))) +
  scale_shape_manual(values=c(15,21,16,17))+
  scale_color_manual(values=c("grey", "dark grey"), guide =FALSE) + 
  guides(shape = guide_legend(override.aes = list(color = "grey", size = 5, linetype =0) ) )


b12.h <- subset(total.sig,  BLOCK == "12" & Sig.SE == TRUE)
b12.c <- subset(B12.pv, QCON < 0.1)

B12.pp <- B12.pv %>%
  mutate(
    Group = case_when(
      SNP %in% singleton$SNP ~ "Singleton",
      SNP %in% b12.c$SNP ~ "Control Treatment Filtered",
      SNP %in% all_con$SNP ~ "Control Treatment Filtered",
      SNP %in% con_filter$SNP ~ "Control Treatment Filtered",
      TRUE ~ "Shared SNP"
    )
  )

man <-ggman(B12.pp, pvalue="QSE", chr="CHR", snp="SNP",bp="BP", relative.positions = TRUE ,sigLine = NA, pointSize=1.5,title = "", alpha = 0.5)

dfm <- man[[1]]

dfm$chrom_alt <- factor(dfm$chrom_alt, levels=c(0,1))

dfmo <- semi_join(dfm, b12.h, by = "SNP")
dfmo$Group <- "Outlier"

dfmsplit <- split(dfm, dfm$chrom)
xbreaks <- sapply(dfmsplit,function(x) {
    midpoint <- length(x$index)/2
    if(midpoint <1) midpoint <- 1
    return(x$index[midpoint])
})

b12.SE <-ggplot(dfm, aes(x= index, y=marker, colour = as.factor(chrom_alt)))+
  geom_point(aes(alpha=marker, shape=Group, size = marker))+  
  geom_point(data=dfmo, fill=cbPaletteSmall4[3], aes(x=index, y=marker, shape=Group, size=marker),  color = cbPaletteSmall4[3])+
  scale_x_continuous(breaks = xbreaks, labels = names(xbreaks),expand = c(0,0),limits = c(0,max(dfm$index)+10)) +
  guides(colour = FALSE,alpha=FALSE, fill = FALSE, size=FALSE) +
  #scale_y_continuous(expand = c(0,0),limits=c(0,max(dfm$marker+1,na.rm=TRUE)))+
  scale_y_continuous(expand = c(0,0),limits=c(0,15))+ #Removes two singleton loci at end of CHR 5
  labs(x = "Chromosome") +
  scale_size_continuous(range = c(0.25,3), limits= c(0,15)) +
  ggtitle("Block 12") +
  theme_classic()+
  labs(y=expression(-Log10(q-value))) +
  scale_shape_manual(values=c(15,21,16,17))+
  scale_color_manual(values=c("grey", "dark grey"), guide =FALSE) +
  guides(shape = guide_legend(override.aes = list(color = "grey", size = 5, linetype =0) ) )

```
```{r}
png(filename="SEmpHighlighted.png", type="cairo",units="px", width=5600, height=3000, res=300, bg="transparent")

(b10.SE / b11.SE /b12.SE )  + plot_annotation(tag_levels = 'A') + plot_layout(guides = 'collect') 
#(b10.SE / b11.SE /b12.SE )  + plot_annotation(tag_levels = 'A') & theme_black()
#man+scale_fill_manual(values = c("#009E73","#E69F00", "#0072B2"),name = "Legend")
null <- dev.off()
```

```{r}
png(filename="SEmpHighlightedBB.png", type="cairo",units="px", width=5600, height=3000, res=300, bg="transparent")

b10.SE.b <- b10.SE +theme_black()
b11.SE.b <- b11.SE +theme_black()
b12.SE.b <- b12.SE +theme_black()

(b10.SE.b / b11.SE.b /b12.SE.b )  + plot_annotation(tag_levels = 'A') + plot_layout(guides = 'collect') & theme_black()
#(b10.SE / b11.SE /b12.SE )  + plot_annotation(tag_levels = 'A') & theme_black()
#man+scale_fill_manual(values = c("#009E73","#E69F00", "#0072B2"),name = "Legend")
null <- dev.off()
```

```{r}
png(filename="SEmpHighlightedBB12.png", type="cairo",units="px", width=5600, height=3000, res=300, bg="transparent")

b12.SE <-ggplot(dfm, aes(x= index, y=marker, colour = as.factor(chrom_alt)))+
  geom_point(aes(alpha=marker, shape=Group, size = marker, show.legend = FALSE))+  
  geom_point(data=dfmo, fill=cbPaletteSmall4[3], aes(x=index, y=marker, shape=Group, size=8),  color = cbPaletteSmall4[3])+
  scale_alpha_continuous(range=c(0.05,0.8), limits= c(0,15))+
  scale_x_continuous(breaks = xbreaks, labels = names(xbreaks),expand = c(0,0),limits = c(0,max(dfm$index)+10)) +
  #scale_y_continuous(expand = c(0,0),limits=c(0,max(dfm$marker+1,na.rm=TRUE)))+
  scale_y_continuous(expand = c(0,0),limits=c(0,15))+ #Removes two singleton loci at end of CHR 5
  labs(x = "Chromosome") +
  theme_black()+
  theme(legend.text = element_text(size = 18), legend.title = element_text(size=24), legend.position = "bottom")+ 
  scale_size_continuous(range = c(0.5,5), limits= c(0,15)) +
  ggtitle("Block 12") +
  labs(y=expression(-Log10(q-value))) +
  scale_shape_manual(values=c(15,21,16,17))+
  scale_color_manual(values=c("grey", "dark grey"))+
  guides(colour = FALSE,alpha=FALSE, fill = FALSE, size=FALSE) +
  guides(shape = guide_legend(override.aes = list(color = "grey", size = 6, linetype =0) ) )
#guides(color = guide_legend(override.aes = list(color = c("grey", "grey"))))
b12.SE
null <- dev.off()
```

```{r}
ball.h <- subset(total.sig, Sig.SE == TRUE)
ball.h <- ball.h %>%
  group_by(SNP) %>%
  slice_min(QSE)

man <-ggman(ball.h, pvalue="QSE", chr="CHR", snp="SNP",bp="BP", relative.positions = TRUE ,sigLine = NA, pointSize=1.5,title = "", alpha = 0.5)

dfm <- man[[1]]

dfm$chrom_alt <- factor(dfm$chrom_alt, levels=c(0,1))

dfmo <- semi_join(dfm, b12.h, by = "SNP")
dfmo$Group <- "Outlier"

dfmsplit <- split(dfm, dfm$chrom)
xbreaks <- sapply(dfmsplit,function(x) {
    midpoint <- length(x$index)/2
    if(midpoint <1) midpoint <- 1
    return(x$index[midpoint])
})

ball.SE <-ggplot(dfm, aes(x= index, y=marker, colour = as.factor(chrom_alt)))+
  geom_point(data=dfm, fill=cbPaletteSmall4[4], aes(x=index, y=marker, size=marker),  color = cbPaletteSmall4[3])+
  scale_x_continuous(breaks = xbreaks, labels = names(xbreaks),expand = c(0,0),limits = c(0,max(dfm$index)+10)) +
  guides(colour = FALSE,alpha=FALSE, fill = FALSE, size=FALSE) +
  scale_y_continuous(expand = c(0,0),limits=c(0,max(dfm$marker+1,na.rm=TRUE)))+
  #scale_y_continuous(expand = c(0,0),limits=c(0,20))+ #Removes two outliers near end of CHR5
  labs(x = "Chromosome") +
  scale_alpha_continuous(range=c(0.05,0.8), limits= c(0,20))+
  scale_size_continuous(range = c(0.25,3), limits=c(0,20)) +
  theme_classic()+
  labs(y=expression(-Log10(q-value))) +
  scale_color_manual(values=c("grey", "dark grey"), guide =FALSE) 

png(filename="SEmpBB.png", type="cairo",units="px", width=4800, height=1000, res=300, bg="transparent")  
ball.SE + theme_black()
null <- dev.off()

```

#### CA
```{r, warning =FALSE, eval = FALSE}
b10.h <- subset(total.sig,  BLOCK == "10" & Sig.CA == TRUE)
b10.c <- subset(B10.pv, QCON < 0.1)

B10.pp <- B10.pv %>%
  mutate(
    Group = case_when(
      SNP %in% singleton$SNP ~ "Singleton",
      SNP %in% b10.c$SNP ~ "Control Treatment Filtered",
      SNP %in% all_con$SNP ~ "Control Treatment Filtered",
      SNP %in% con_filter$SNP ~ "Control Treatment Filtered",
      TRUE ~ "Shared SNP"
    )
  )

man <-ggman(B10.pp, pvalue="QCA", chr="CHR", snp="SNP",bp="BP", relative.positions = TRUE ,sigLine = NA, pointSize=1.5,title = "", alpha = 0.5)

dfm <- man[[1]]

dfm$chrom_alt <- factor(dfm$chrom_alt, levels=c(0,1))

dfmo <- semi_join(dfm, b10.h, by = "SNP")
dfmo$Group <- "Outlier"

dfmsplit <- split(dfm, dfm$chrom)
xbreaks <- sapply(dfmsplit,function(x) {
    midpoint <- length(x$index)/2
    if(midpoint <1) midpoint <- 1
    return(x$index[midpoint])
})

b10.CA <-ggplot(dfm, aes(x= index, y=marker, colour = as.factor(chrom_alt)))+
  geom_point(aes(alpha=marker, shape=Group, size = marker))+  
  geom_point(data=dfmo, fill=cbPaletteSmall4[2], aes(x=index, y=marker, shape=Group, size=marker),  color = cbPaletteSmall4[2])+
  scale_x_continuous(breaks = xbreaks, labels = names(xbreaks),expand = c(0,0),limits = c(0,max(dfm$index)+10)) +
  guides(colour = FALSE,alpha=FALSE, fill = FALSE, size=FALSE) +
  #scale_y_continuous(expand = c(0,0),limits=c(0,max(dfm$marker+1,na.rm=TRUE)))+
  scale_y_continuous(expand = c(0,0),limits=c(0,32))+ #Limit manually set, removes 3 control filtered loci in CHR 2
  ggtitle("Block 10") +
  labs(x = "Chromosome") +
  scale_size_continuous(range = c(0.25,3), limits=c(0,32)) +
  theme_classic()+
  labs(y=expression(-Log10(q-value))) +
  scale_shape_manual(values=c(15,21,16,17))+
  scale_color_manual(values=c("grey", "dark grey"), guide =FALSE) +
  guides(shape = guide_legend(override.aes = list(color = "grey", size = 5, linetype =0) ) )

b11.h <- subset(total.sig,  BLOCK == "11" & Sig.CA == TRUE)
b11.c <- subset(B11.pv, QCON < 0.1)

B11.pp <- B11.pv %>%
  mutate(
    Group = case_when(
      SNP %in% singleton$SNP ~ "Singleton",
      SNP %in% b11.c$SNP ~ "Control Treatment Filtered",
      SNP %in% all_con$SNP ~ "Control Treatment Filtered",
      SNP %in% con_filter$SNP ~ "Control Treatment Filtered",
      TRUE ~ "Shared SNP"
    )
  )

man <-ggman(B11.pp, pvalue="QCA", chr="CHR", snp="SNP",bp="BP", relative.positions = TRUE ,sigLine = NA, pointSize=1.5,title = "", alpha = 0.5)

dfm <- man[[1]]

dfm$chrom_alt <- factor(dfm$chrom_alt, levels=c(0,1))


dfmo <- semi_join(dfm, b11.h, by = "SNP")
dfmo$Group <- "Outlier"

dfmsplit <- split(dfm, dfm$chrom)
xbreaks <- sapply(dfmsplit,function(x) {
    midpoint <- length(x$index)/2
    if(midpoint <1) midpoint <- 1
    return(x$index[midpoint])
})

b11.CA <-ggplot(dfm, aes(x= index, y=marker, colour = as.factor(chrom_alt)))+
  geom_point(aes(alpha=marker, shape=Group, size = marker))+
  #geom_point(aes(x=index, y=log10(QCON), colour = as.factor(chrom_alt), alpha=0.5, shape = Group))+  
  geom_point(data=dfmo, fill=cbPaletteSmall4[2], aes(x=index, y=marker, shape=Group, size=marker),  color = cbPaletteSmall4[2])+
  scale_x_continuous(breaks = xbreaks, labels = names(xbreaks),expand = c(0,0),limits = c(0,max(dfm$index)+10)) +
  guides(colour = FALSE,alpha=FALSE, fill = FALSE, size=FALSE) +
  scale_y_continuous(expand = c(0,0),limits=c(0,max(dfm$marker+1,na.rm=TRUE)))+
  labs(x = "Chromosome") +
  ggtitle("Block 11") +
  scale_size_continuous(range = c(0.25,3)) +
  theme_classic()+
  labs(y=expression(-Log10(q-value))) +
  scale_shape_manual(values=c(15,21,16,17))+
  scale_color_manual(values=c("grey", "dark grey"), guide =FALSE) + 
  guides(shape = guide_legend(override.aes = list(color = "grey", size = 5, linetype =0) ) )


b12.h <- subset(total.sig,  BLOCK == "12" & Sig.CA == TRUE)
b12.c <- subset(B12.pv, QCON < 0.1)

B12.pp <- B12.pv %>%
  mutate(
    Group = case_when(
      SNP %in% singleton$SNP ~ "Singleton",
      SNP %in% b12.c$SNP ~ "Control Treatment Filtered",
      SNP %in% all_con$SNP ~ "Control Treatment Filtered",
      SNP %in% con_filter$SNP ~ "Control Treatment Filtered",
      TRUE ~ "Shared SNP"
    )
  )

man <-ggman(B12.pp, pvalue="QCA", chr="CHR", snp="SNP",bp="BP", relative.positions = TRUE ,sigLine = NA, pointSize=1.5,title = "", alpha = 0.5)

dfm <- man[[1]]

dfm$chrom_alt <- factor(dfm$chrom_alt, levels=c(0,1))

dfmo <- semi_join(dfm, b12.h, by = "SNP")
dfmo$Group <- "Outlier"

dfmsplit <- split(dfm, dfm$chrom)
xbreaks <- sapply(dfmsplit,function(x) {
    midpoint <- length(x$index)/2
    if(midpoint <1) midpoint <- 1
    return(x$index[midpoint])
})

b12.CA <-ggplot(dfm, aes(x= index, y=marker, colour = as.factor(chrom_alt)))+
  geom_point(aes(alpha=marker, shape=Group, size = marker))+ 
  geom_point(data=dfmo, fill=cbPaletteSmall4[2], aes(x=index, y=marker, shape=Group, size=marker),  color = cbPaletteSmall4[2])+
  scale_x_continuous(breaks = xbreaks, labels = names(xbreaks),expand = c(0,0),limits = c(0,max(dfm$index)+10)) +
  guides(colour = FALSE,alpha=FALSE, fill = FALSE, size=FALSE) +
  #scale_y_continuous(expand = c(0,0),limits=c(0,max(dfm$marker+1,na.rm=TRUE)))+
  scale_y_continuous(expand = c(0,0),limits=c(0,20))+ #removes two singleton loci near end of CHR 5
  labs(x = "Chromosome") +
  scale_size_continuous(range = c(0.25,3), limits=c(0,20)) +
  ggtitle("Block 12") +
  theme_classic()+
  labs(y=expression(-Log10(q-value))) +
  scale_shape_manual(values=c(15,21,16,17))+
  scale_color_manual(values=c("grey", "dark grey"), guide =FALSE) 
```
```{r}
png(filename="CAmpHighlighted.png", type="cairo",units="px", width=5600, height=3000, res=300, bg="transparent")

(b10.CA / b11.CA /b12.CA )  + plot_annotation(tag_levels = 'A') + plot_layout(guides = 'collect') 
#(b10.CA / b11.CA /b12.CA )  + plot_annotation(tag_levels = 'A') & theme_black()
#man+scale_fill_manual(values = c("#009E73","#E69F00", "#0072B2"),name = "Legend")
null <- dev.off()
```

```{r}
png(filename="CAmpHighlightedBB.png", type="cairo",units="px", width=5600, height=3000, res=300, bg="transparent")

b10.CA.b <- b10.CA +theme_black()
b11.CA.b <- b11.CA +theme_black()
b12.CA.b <- b12.CA +theme_black()

(b10.CA.b / b11.CA.b /b12.CA.b )  + plot_annotation(tag_levels = 'A') + plot_layout(guides = 'collect') & theme_black()
#(b10.SE / b11.SE /b12.SE )  + plot_annotation(tag_levels = 'A') & theme_black()
#man+scale_fill_manual(values = c("#009E73","#E69F00", "#0072B2"),name = "Legend")
null <- dev.off()
```


```{r}
ball.h <- subset(total.sig, Sig.CA == TRUE)
ball.h <- ball.h %>%
  group_by(SNP) %>%
  slice_min(QCA)

man <-ggman(ball.h, pvalue="QSE", chr="CHR", snp="SNP",bp="BP", relative.positions = TRUE ,sigLine = NA, pointSize=1.5,title = "", alpha = 0.5)

dfm <- man[[1]]

dfm$chrom_alt <- factor(dfm$chrom_alt, levels=c(0,1))

dfmo <- semi_join(dfm, b12.h, by = "SNP")
dfmo$Group <- "Outlier"

dfmsplit <- split(dfm, dfm$chrom)
xbreaks <- sapply(dfmsplit,function(x) {
    midpoint <- length(x$index)/2
    if(midpoint <1) midpoint <- 1
    return(x$index[midpoint])
})

ball.CA <-ggplot(dfm, aes(x= index, y=marker, colour = as.factor(chrom_alt)))+
  geom_point(data=dfm, fill=cbPaletteSmall4[4], aes(x=index, y=marker, size=marker),  color = cbPaletteSmall4[2])+
  scale_x_continuous(breaks = xbreaks, labels = names(xbreaks),expand = c(0,0),limits = c(0,max(dfm$index)+10)) +
  guides(colour = FALSE,alpha=FALSE, fill = FALSE, size=FALSE) +
  scale_y_continuous(expand = c(0,0),limits=c(0,max(dfm$marker+1,na.rm=TRUE)))+
  #scale_y_continuous(expand = c(0,0),limits=c(0,20))+ #Removes two outliers near end of CHR5
  labs(x = "Chromosome") +
  scale_alpha_continuous(range=c(0.05,0.8), limits= c(0,20))+
  scale_size_continuous(range = c(0.25,3), limits=c(0,20)) +
  theme_classic()+
  labs(y=expression(-Log10(q-value))) +
  scale_color_manual(values=c("grey", "dark grey"), guide =FALSE) 

png(filename="CAmpBB.png", type="cairo",units="px", width=4800, height=1000, res=300, bg="transparent")  
ball.CA + theme_black()
null <- dev.off()

```

# Venn Diagrams
```{r}
library(VennDiagram)

sig.CA <- nrow(unique(subset(grouped_total.sig, Sig.CA == "TRUE",c(SNP))))
sig.SE <- nrow(unique(subset(grouped_total.sig, Sig.SE == "TRUE",c(SNP))))
sig.CASE <- nrow(unique(subset(grouped_total.sig, Sig.CASE == "TRUE",c(SNP))))

sig.CA.SE <- nrow(unique(subset(grouped_total.sig, Sig.CA == "TRUE" & Sig.SE == "TRUE",c(SNP) )))
sig.CA.CASE <- nrow(unique(subset(grouped_total.sig,Sig.CA == "TRUE" & Sig.CASE == "TRUE",c(SNP) )))

sig.SE.CASE <- nrow(unique(subset(grouped_total.sig, Sig.CASE == "TRUE" & Sig.SE == "TRUE" ,c(SNP))))

sig.all <- nrow(unique(subset(grouped_total.sig, Sig.CASE == "TRUE" & Sig.CA == "TRUE" & Sig.SE == "TRUE",c(SNP))))
#sig.all <- nrow(subset(pp1, Sig.CA == TRUE & Sig.SE == TRUE  & Sig.CASE == TRUE & QPCADAPT < alpha))

png(filename="Ven.png", type="cairo",units="px", width=3000, height=3000, res=200, bg="transparent")

venn.plot <- draw.triple.venn(
  area1=sig.CA, area2=sig.SE, area3=sig.CASE, 
  n12= sig.CA.SE, n13=sig.CA.CASE, 
  n23=sig.SE.CASE,
  n123=sig.all, cex=5,
  category = c("CA", "SE", "CASE"),
  fill = cbPaletteSmall3,cat.cex = rep(5, 3),
  cat.col = cbPaletteSmall3, label.col = rep("white", 7))

dev.off()
grid.draw(venn.plot)
grid.newpage()

png(filename="VenBO.png", type="cairo",units="px", width=3000, height=3000, res=200, bg="transparent")
venn.plot <- draw.triple.venn(
  area1=sig.CA, area2=sig.SE, area3=sig.CASE, 
  n12= sig.CA.SE, n13=sig.CA.CASE, 
  n23=sig.SE.CASE,
  n123=sig.all, cex=5,
  category = c("CA", "SE", "CASE"),
  fill = cbPaletteSmall3,cat.cex = rep(5, 3),
  cat.col = cbPaletteSmall3, label.col = rep("black", 7))
dev.off()
```
### B10 Only
```{r}
sig.CA <- nrow(unique(subset(ppsig.B10.01, Sig.CA == TRUE,c(SNP))))
sig.SE <- nrow(unique(subset(ppsig.B10.01, Sig.SE == TRUE ,c(SNP))))
sig.CASE <- nrow(unique(subset(ppsig.B10.01, Sig.CASE == TRUE,c(SNP))))

sig.CA.SE <- nrow(unique(subset(ppsig.B10.01,Sig.CA == TRUE & Sig.SE == TRUE,c(SNP) )))
sig.CA.CASE <- nrow(unique(subset(ppsig.B10.01,Sig.CA == TRUE & Sig.CASE == TRUE ,c(SNP) )))

sig.SE.CASE <- nrow(unique(subset(ppsig.B10.01, Sig.SE == TRUE & Sig.CASE == TRUE ,c(SNP))))

sig.all <- nrow(unique(subset(ppsig.B10.01, Sig.CA == TRUE & Sig.SE == TRUE & Sig.CASE == TRUE,c(SNP))))
#sig.all <- nrow(subset(pp1, Sig.CA == TRUE & Sig.SE == TRUE  & Sig.CASE == TRUE & QPCADAPT < alpha))

png(filename="VenB10.png", type="cairo",units="px", width=3000, height=3000, res=200, bg="transparent")

venn.plot <- draw.triple.venn(
  area1=sig.CA, area2=sig.SE, area3=sig.CASE, 
  n12= sig.CA.SE, n13=sig.CA.CASE, 
  n23=sig.SE.CASE,
  n123=sig.all, cex=5,
  category = c("CA", "SE", "CASE"),
  fill = cbPaletteSmall3,cat.cex = rep(5, 3),
  cat.col = cbPaletteSmall3, label.col = rep("white", 7))
dev.off()
png(filename="VenBOB10.png", type="cairo",units="px", width=3000, height=3000, res=200, bg="transparent")
venn.plot <- draw.triple.venn(
  area1=sig.CA, area2=sig.SE, area3=sig.CASE, 
  n12= sig.CA.SE, n13=sig.CA.CASE, 
  n23=sig.SE.CASE,
  n123=sig.all, cex=5,
  category = c("CA", "SE", "CASE"),
  fill = cbPaletteSmall3,cat.cex = rep(5, 3),
  cat.col = cbPaletteSmall3, label.col = rep("black", 7))
dev.off()
```
### B11 Only
```{r}
sig.CA <- nrow(unique(subset(ppsig.B11.01, Sig.CA == TRUE,c(SNP))))
sig.SE <- nrow(unique(subset(ppsig.B11.01, Sig.SE == TRUE,c(SNP))))
sig.CASE <- nrow(unique(subset(ppsig.B11.01, Sig.CASE == TRUE,c(SNP))))

sig.CA.SE <- nrow(unique(subset(ppsig.B11.01, Sig.CA == TRUE & Sig.SE == TRUE,c(SNP) )))
sig.CA.CASE <- nrow(unique(subset(ppsig.B11.01,Sig.CA == TRUE & Sig.CASE == TRUE ,c(SNP) )))

sig.SE.CASE <- nrow(unique(subset(ppsig.B11.01, Sig.SE == TRUE & Sig.CASE == TRUE ,c(SNP))))

sig.all <- nrow(unique(subset(ppsig.B11.01, Sig.CA == TRUE & Sig.SE == TRUE & Sig.CASE == TRUE,c(SNP))))
#sig.all <- nrow(subset(pp1, Sig.CA == TRUE & Sig.SE == TRUE  & Sig.CASE == TRUE & QPCADAPT < alpha))

png(filename="VenB11.png", type="cairo",units="px", width=3000, height=3000, res=200, bg="transparent")

venn.plot <- draw.triple.venn(
  area1=sig.CA, area2=sig.SE, area3=sig.CASE, 
  n12= sig.CA.SE, n13=sig.CA.CASE, 
  n23=sig.SE.CASE,
  n123=sig.all, cex=5,
  category = c("CA", "SE", "CASE"),
  fill = cbPaletteSmall3,cat.cex = rep(5, 3),
  cat.col = cbPaletteSmall3, label.col = rep("white", 7))
dev.off()
png(filename="VenBOB11.png", type="cairo",units="px", width=3000, height=3000, res=200, bg="transparent")
venn.plot <- draw.triple.venn(
  area1=sig.CA, area2=sig.SE, area3=sig.CASE, 
  n12= sig.CA.SE, n13=sig.CA.CASE, 
  n23=sig.SE.CASE,
  n123=sig.all, cex=5,
  category = c("CA", "SE", "CASE"),
  fill = cbPaletteSmall3,cat.cex = rep(5, 3),
  cat.col = cbPaletteSmall3, label.col = rep("black", 7))
dev.off()
```

### B12 Only
```{r}
library(VennDiagram)
alpha = 0.01
alpha2 = 0.1

#pp2 <- subset(pp1, QCON > alpha2)
sig.CA <- nrow(unique(subset(ppsig.B11.01, Sig.CA == TRUE,c(SNP))))
sig.SE <- nrow(unique(subset(ppsig.B11.01, Sig.SE == TRUE,c(SNP))))
sig.CASE <- nrow(unique(subset(ppsig.B11.01, Sig.CASE == TRUE,c(SNP))))

sig.CA.SE <- nrow(unique(subset(ppsig.B11.01, Sig.CA == TRUE & Sig.SE == TRUE,c(SNP) )))
sig.CA.CASE <- nrow(unique(subset(ppsig.B11.01,Sig.CA == TRUE & Sig.CASE == TRUE ,c(SNP) )))

sig.SE.CASE <- nrow(unique(subset(ppsig.B11.01, Sig.SE == TRUE & Sig.CASE == TRUE ,c(SNP))))

sig.all <- nrow(unique(subset(ppsig.B11.01, Sig.CA == TRUE & Sig.SE == TRUE & Sig.CASE == TRUE,c(SNP))))

png(filename="VenB12.png", type="cairo",units="px", width=3000, height=3000, res=200, bg="transparent")

venn.plot <- draw.triple.venn(
  area1=sig.CA, area2=sig.SE, area3=sig.CASE, 
  n12= sig.CA.SE, n13=sig.CA.CASE, 
  n23=sig.SE.CASE,
  n123=sig.all, cex=5,
  category = c("CA", "SE", "CASE"),
  fill = cbPaletteSmall3,cat.cex = rep(5, 3),
  cat.col = cbPaletteSmall3, label.col = rep("white", 7))
dev.off()
png(filename="VenBOB12.png", type="cairo",units="px", width=3000, height=3000, res=200, bg="transparent")
venn.plot <- draw.triple.venn(
  area1=sig.CA, area2=sig.SE, area3=sig.CASE, 
  n12= sig.CA.SE, n13=sig.CA.CASE, 
  n23=sig.SE.CASE,
  n123=sig.all, cex=5,
  category = c("CA", "SE", "CASE"),
  fill = cbPaletteSmall3,cat.cex = rep(5, 3),
  cat.col = cbPaletteSmall3, label.col = rep("black", 7))
dev.off()
```

### LOC Only
```{bash}
source activate CASE

cat Sig.loci.1.CA.LOC Sig.loci.1.SE.LOC | sort | uniq -c | mawk '$1 > 1'  | mawk '{print $2}' > Sig.loci.1.CA.SE.LOC
cat Sig.loci.1.CA.LOC Sig.loci.1.CASE.LOC | sort | uniq -c | mawk '$1 > 1'| mawk '{print $2}' > Sig.loci.1.CA.CASE.LOC
cat Sig.loci.1.SE.LOC Sig.loci.1.CASE.LOC | sort | uniq -c | mawk '$1 > 1' | mawk '{print $2}'> Sig.loci.1.SE.CASE.LOC
cat Sig.loci.1.SE.LOC Sig.loci.1.CASE.LOC Sig.loci.1.CA.LOC | sort | uniq -c | mawk '$1 > 2'| mawk '{print $2}' > Sig.loci.1.SE.CASE.CA.LOC

cat Sig.loci.1.CASE.LOC <(cat Sig.loci.1.SE.LOC Sig.loci.1.CASE.LOC Sig.loci.1.CA.LOC | sort | uniq -c | mawk '$1 < 2' | mawk '{print $2}') | sort | uniq -c | mawk '$1 > 1' | mawk '{print $2}' > Sig.loci.CASE.ONLY.LOC
cat Sig.loci.1.SE.LOC <(cat Sig.loci.1.SE.LOC Sig.loci.1.CASE.LOC Sig.loci.1.CA.LOC | sort | uniq -c | mawk '$1 < 2' | mawk '{print $2}') | sort | uniq -c | mawk '$1 > 1' | mawk '{print $2}' > Sig.loci.SE.ONLY.LOC
cat Sig.loci.1.CA.LOC <(cat Sig.loci.1.SE.LOC Sig.loci.1.CASE.LOC Sig.loci.1.CA.LOC | sort | uniq -c | mawk '$1 < 2' | mawk '{print $2}') | sort | uniq -c | mawk '$1 > 1' | mawk '{print $2}' > Sig.loci.CA.ONLY.LOC

grep -v -f Sig.loci.1.CA.LOC Sig.loci.1.SE.CASE.LOC > Sig.loci.CASE.SE.ONLY.LOC



cat Sig.loci.B10.CA.LOC Sig.loci.B10.SE.LOC | sort | uniq -c | mawk '$1 > 1'  | mawk '{print $2}' > Sig.loci.B10.CA.SE.LOC
cat Sig.loci.B10.CA.LOC Sig.loci.B10.CASE.LOC | sort | uniq -c | mawk '$1 > 1'| mawk '{print $2}' > Sig.loci.B10.CA.CASE.LOC
cat Sig.loci.B10.SE.LOC Sig.loci.B10.CASE.LOC | sort | uniq -c | mawk '$1 > 1' | mawk '{print $2}'> Sig.loci.B10.SE.CASE.LOC
cat Sig.loci.B10.SE.LOC Sig.loci.B10.CASE.LOC Sig.loci.B10.CA.LOC | sort | uniq -c | mawk '$1 > 2'| mawk '{print $2}' > Sig.loci.B10.SE.CASE.CA.LOC

cat Sig.loci.B10.CASE.LOC <(cat Sig.loci.B10.SE.LOC Sig.loci.B10.CASE.LOC Sig.loci.B10.CA.LOC | sort | uniq -c | mawk '$1 < 2' | mawk '{print $2}') | sort | uniq -c | mawk '$1 > 1' | mawk '{print $2}' > Sig.loci.B10.CASE.ONLY.LOC
cat Sig.loci.B10.SE.LOC <(cat Sig.loci.B10.SE.LOC Sig.loci.B10.CASE.LOC Sig.loci.B10.CA.LOC | sort | uniq -c | mawk '$1 < 2' | mawk '{print $2}') | sort | uniq -c | mawk '$1 > 1' | mawk '{print $2}' > Sig.loci.B10.SE.ONLY.LOC
cat Sig.loci.B10.CA.LOC <(cat Sig.loci.B10.SE.LOC Sig.loci.B10.CASE.LOC Sig.loci.B10.CA.LOC | sort | uniq -c | mawk '$1 < 2' | mawk '{print $2}') | sort | uniq -c | mawk '$1 > 1' | mawk '{print $2}' > Sig.loci.B10.CA.ONLY.LOC

grep -v -f Sig.loci.B10.CA.LOC Sig.loci.B10.SE.CASE.LOC > Sig.loci.B10.CASE.SE.ONLY.LOC

cat Sig.loci.B11.CA.LOC Sig.loci.B11.SE.LOC | sort | uniq -c | mawk '$1 > 1'  | mawk '{print $2}' > Sig.loci.B11.CA.SE.LOC
cat Sig.loci.B11.CA.LOC Sig.loci.B11.CASE.LOC | sort | uniq -c | mawk '$1 > 1'| mawk '{print $2}' > Sig.loci.B11.CA.CASE.LOC
cat Sig.loci.B11.SE.LOC Sig.loci.B11.CASE.LOC | sort | uniq -c | mawk '$1 > 1' | mawk '{print $2}'> Sig.loci.B11.SE.CASE.LOC
cat Sig.loci.B11.SE.LOC Sig.loci.B11.CASE.LOC Sig.loci.B11.CA.LOC | sort | uniq -c | mawk '$1 > 2'| mawk '{print $2}' > Sig.loci.B11.SE.CASE.CA.LOC

cat Sig.loci.B11.CASE.LOC <(cat Sig.loci.B11.SE.LOC Sig.loci.B11.CASE.LOC Sig.loci.B11.CA.LOC | sort | uniq -c | mawk '$1 < 2' | mawk '{print $2}') | sort | uniq -c | mawk '$1 > 1' | mawk '{print $2}' > Sig.loci.B11.CASE.ONLY.LOC
cat Sig.loci.B11.SE.LOC <(cat Sig.loci.B11.SE.LOC Sig.loci.B11.CASE.LOC Sig.loci.B11.CA.LOC | sort | uniq -c | mawk '$1 < 2' | mawk '{print $2}') | sort | uniq -c | mawk '$1 > 1' | mawk '{print $2}' > Sig.loci.B11.SE.ONLY.LOC
cat Sig.loci.B11.CA.LOC <(cat Sig.loci.B11.SE.LOC Sig.loci.B11.CASE.LOC Sig.loci.B11.CA.LOC | sort | uniq -c | mawk '$1 < 2' | mawk '{print $2}') | sort | uniq -c | mawk '$1 > 1' | mawk '{print $2}' > Sig.loci.B11.CA.ONLY.LOC

grep -v -f Sig.loci.B11.CA.LOC Sig.loci.B11.SE.CASE.LOC > Sig.loci.B11.CASE.SE.ONLY.LOC

cat Sig.loci.B12.CA.LOC Sig.loci.B12.SE.LOC | sort | uniq -c | mawk '$1 > 1'  | mawk '{print $2}' > Sig.loci.B12.CA.SE.LOC
cat Sig.loci.B12.CA.LOC Sig.loci.B12.CASE.LOC | sort | uniq -c | mawk '$1 > 1'| mawk '{print $2}' > Sig.loci.B12.CA.CASE.LOC
cat Sig.loci.B12.SE.LOC Sig.loci.B12.CASE.LOC | sort | uniq -c | mawk '$1 > 1' | mawk '{print $2}'> Sig.loci.B12.SE.CASE.LOC
cat Sig.loci.B12.SE.LOC Sig.loci.B12.CASE.LOC Sig.loci.B12.CA.LOC | sort | uniq -c | mawk '$1 > 2'| mawk '{print $2}' > Sig.loci.B12.SE.CASE.CA.LOC

cat Sig.loci.B12.CASE.LOC <(cat Sig.loci.B12.SE.LOC Sig.loci.B12.CASE.LOC Sig.loci.B12.CA.LOC | sort | uniq -c | mawk '$1 < 2' | mawk '{print $2}') | sort | uniq -c | mawk '$1 > 1' | mawk '{print $2}' > Sig.loci.B12.CASE.ONLY.LOC
cat Sig.loci.B12.SE.LOC <(cat Sig.loci.B12.SE.LOC Sig.loci.B12.CASE.LOC Sig.loci.B12.CA.LOC | sort | uniq -c | mawk '$1 < 2' | mawk '{print $2}') | sort | uniq -c | mawk '$1 > 1' | mawk '{print $2}' > Sig.loci.B12.SE.ONLY.LOC
cat Sig.loci.B12.CA.LOC <(cat Sig.loci.B12.SE.LOC Sig.loci.B12.CASE.LOC Sig.loci.B12.CA.LOC | sort | uniq -c | mawk '$1 < 2' | mawk '{print $2}') | sort | uniq -c | mawk '$1 > 1' | mawk '{print $2}' > Sig.loci.B12.CA.ONLY.LOC

grep -v -f Sig.loci.B12.CA.LOC Sig.loci.B12.SE.CASE.LOC > Sig.loci.B12.CASE.SE.ONLY.LOC

```

```{r}
library(VennDiagram)
alpha = 0.05
alpha2 = 0.1

Sig.loci.1.CA.LOC <- read.table("Sig.loci.1.CA.LOC", header =FALSE)
Sig.loci.1.SE.LOC <- read.table("Sig.loci.1.SE.LOC", header =FALSE)
Sig.loci.1.CASE.LOC <- read.table("Sig.loci.1.CASE.LOC", header =FALSE)

sig.CA <- nrow(unique(Sig.loci.1.CA.LOC))
sig.SE <- nrow(unique(Sig.loci.1.SE.LOC))
sig.CASE <- nrow(unique(Sig.loci.1.CASE.LOC))

sig.CA.SE <- nrow(unique(read.table("Sig.loci.1.CA.SE.LOC", header =FALSE)))
sig.CA.CASE <- nrow(unique(read.table("Sig.loci.1.CA.CASE.LOC", header =FALSE)))
sig.SE.CASE <- nrow(unique(read.table("Sig.loci.1.SE.CASE.LOC", header =FALSE)))

sig.all <- nrow(unique(read.table("Sig.loci.1.SE.CASE.CA.LOC", header =FALSE)))


png(filename="VenLOC.png", type="cairo",units="px", width=3000, height=3000, res=200, bg="transparent")
venn.plot <- draw.triple.venn(
  area1=sig.CA, area2=sig.SE, area3=sig.CASE, 
  n12= sig.CA.SE, n13=sig.CA.CASE, 
  n23=sig.SE.CASE,
  n123=sig.all, cex=5,
  category = c("CA", "SE", "CASE"),
  fill = cbPaletteSmall3,cat.cex = rep(5, 3),
  cat.col = cbPaletteSmall3, label.col = rep("white", 7))
dev.off()
grid.draw(venn.plot)
grid.newpage()
```
### B10
```{r}
library(VennDiagram)
alpha = 0.05
alpha2 = 0.1

Sig.loci.B10.CA.LOC <- read.table("Sig.loci.B10.CA.LOC", header =FALSE)
Sig.loci.B10.SE.LOC <- read.table("Sig.loci.B10.SE.LOC", header =FALSE)
Sig.loci.B10.CASE.LOC <- read.table("Sig.loci.B10.CASE.LOC", header =FALSE)

sig.CA <- nrow(unique(Sig.loci.B10.CA.LOC))
sig.SE <- nrow(unique(Sig.loci.B10.SE.LOC))
sig.CASE <- nrow(unique(Sig.loci.B10.CASE.LOC))

sig.CA.SE <- nrow(unique(read.table("Sig.loci.B10.CA.SE.LOC", header =FALSE)))
sig.CA.CASE <- nrow(unique(read.table("Sig.loci.B10.CA.CASE.LOC", header =FALSE)))
sig.SE.CASE <- nrow(unique(read.table("Sig.loci.B10.SE.CASE.LOC", header =FALSE)))

sig.all <- nrow(unique(read.table("Sig.loci.B10.SE.CASE.CA.LOC", header =FALSE)))


png(filename="VenB10LOC.png", type="cairo",units="px", width=3000, height=3000, res=200, bg="transparent")
venn.plot <- draw.triple.venn(
  area1=sig.CA, area2=sig.SE, area3=sig.CASE, 
  n12= sig.CA.SE, n13=sig.CA.CASE, 
  n23=sig.SE.CASE,
  n123=sig.all, cex=5,
  category = c("CA", "SE", "CASE"),
  fill = cbPaletteSmall3,cat.cex = rep(5, 3),
  cat.col = cbPaletteSmall3, label.col = rep("white", 7))
dev.off()

```

### B11
```{r}
library(VennDiagram)
alpha = 0.05
alpha2 = 0.1

Sig.loci.B11.CA.LOC <- read.table("Sig.loci.B11.CA.LOC", header =FALSE)
Sig.loci.B11.SE.LOC <- read.table("Sig.loci.B11.SE.LOC", header =FALSE)
Sig.loci.B11.CASE.LOC <- read.table("Sig.loci.B11.CASE.LOC", header =FALSE)

sig.CA <- nrow(unique(Sig.loci.B11.CA.LOC))
sig.SE <- nrow(unique(Sig.loci.B11.SE.LOC))
sig.CASE <- nrow(unique(Sig.loci.B11.CASE.LOC))

sig.CA.SE <- nrow(unique(read.table("Sig.loci.B11.CA.SE.LOC", header =FALSE)))
sig.CA.CASE <- nrow(unique(read.table("Sig.loci.B11.CA.CASE.LOC", header =FALSE)))
sig.SE.CASE <- nrow(unique(read.table("Sig.loci.B11.SE.CASE.LOC", header =FALSE)))

sig.all <- nrow(unique(read.table("Sig.loci.B11.SE.CASE.CA.LOC", header =FALSE)))


png(filename="VenB11LOC.png", type="cairo",units="px", width=3000, height=3000, res=200, bg="transparent")
venn.plot <- draw.triple.venn(
  area1=sig.CA, area2=sig.SE, area3=sig.CASE, 
  n12= sig.CA.SE, n13=sig.CA.CASE, 
  n23=sig.SE.CASE,
  n123=sig.all, cex=5,
  category = c("CA", "SE", "CASE"),
  fill = cbPaletteSmall3,cat.cex = rep(5, 3),
  cat.col = cbPaletteSmall3, label.col = rep("white", 7))
dev.off()

```

### B12
```{r}
library(VennDiagram)
alpha = 0.05
alpha2 = 0.1

Sig.loci.B12.CA.LOC <- read.table("Sig.loci.B12.CA.LOC", header =FALSE)
Sig.loci.B12.SE.LOC <- read.table("Sig.loci.B12.SE.LOC", header =FALSE)
Sig.loci.B12.CASE.LOC <- read.table("Sig.loci.B12.CASE.LOC", header =FALSE)

sig.CA <- nrow(unique(Sig.loci.B12.CA.LOC))
sig.SE <- nrow(unique(Sig.loci.B12.SE.LOC))
sig.CASE <- nrow(unique(Sig.loci.B12.CASE.LOC))

sig.CA.SE <- nrow(unique(read.table("Sig.loci.B12.CA.SE.LOC", header =FALSE)))
sig.CA.CASE <- nrow(unique(read.table("Sig.loci.B12.CA.CASE.LOC", header =FALSE)))
sig.SE.CASE <- nrow(unique(read.table("Sig.loci.B12.SE.CASE.LOC", header =FALSE)))

sig.all <- nrow(unique(read.table("Sig.loci.B12.SE.CASE.CA.LOC", header =FALSE)))


png(filename="VenB12LOC.png", type="cairo",units="px", width=3000, height=3000, res=200, bg="transparent")
venn.plot <- draw.triple.venn(
  area1=sig.CA, area2=sig.SE, area3=sig.CASE, 
  n12= sig.CA.SE, n13=sig.CA.CASE, 
  n23=sig.SE.CASE,
  n123=sig.all, cex=5,
  category = c("CA", "SE", "CASE"),
  fill = cbPaletteSmall3,cat.cex = rep(5, 3),
  cat.col = cbPaletteSmall3, label.col = rep("white", 7))
dev.off()

```
# PCA
```{bash, eval = FALSE}
source activate CASE

mawk '!/CHR/' Total.Significant.Loci | cut -f2,3 | sort | uniq > TSL.pca.filter

mawk 'NR==FNR{a[$1,$2]; next} ($1,$2) in a' TSL.pca.filter CASE.All.Blocks.cov.sync | cut --complement -f60- > outlier.pca.sync
```

```{r}
input.all.outlier <- read.sync(file="outlier.pca.sync", gen=seq(1,56), repl=seq(1,56), polarization="rising")


input.all.outlier.af <- af(input.all.outlier,gen=seq(1,56), repl=seq(1,56))
treatment.labels <- c(rep("CA", 11),rep("CASE", 11),rep("CON", 11),rep("IS", 12),rep("SE", 11))
spawn.names <- c("B12","B11","B12","B10","B11","B10","B11","B12","B10","B12","B11","B10","B11","B12","B12","B10","B12","B11","B10","B11","B11","B12","B10","B11","B12","B11","B10","B12","B10","B11","B12","B11","B12","B10","B11","B10","B11","B10","B11","B10","B11","B12","B12","B12","B12","B11","B12","B10","B10","B11","B11","B10","B12","B12","B11","B12")

colnames(input.all.outlier.af) <- paste(treatment.labels,spawn.names,sep="_")


# Perform PCA
af_t <- t(input.all.outlier.af)

pca_res <- prcomp(af_t, center = TRUE, scale. = TRUE)

# Get scores for plotting
pca_df <- as.data.frame(pca_res$x)
#pca_df$locus <- rownames(input.all.outlier.af)  # keep locus IDs if you have them

var_explained <- pca_res$sdev^2 / sum(pca_res$sdev^2) * 100

# Example: if you have a vector "pop_group" the same length as rows

treatment.labels <- c(rep("CA", 11),rep("CASE", 11),rep("CON", 11),rep("IS", 12),rep("SE", 11))
spawn.names <- c("B12","B11","B12","B10","B11","B10","B11","B12","B10","B12","B11","B10","B11","B12","B12","B10","B12","B11","B10","B11","B11","B12","B10","B11","B12","B11","B10","B12","B10","B11","B12","B11","B12","B10","B11","B10","B11","B10","B11","B10","B11","B12","B12","B12","B12","B11","B12","B10","B10","B11","B11","B10","B12","B12","B11","B12")

pca_df$Population <- spawn.names
pca_df$Treatments <- treatment.labels

ggplot(pca_df, aes(x = PC1, y = PC2, color = Population, fill=Population)) +
  geom_point(aes(alpha=0.2), size =10,shape=21, col="black")+  scale_fill_manual(values=cbPaletteSmall,name="Treatment")+ 
guides(alpha=FALSE) +theme(axis.title.x = element_text(size = 24),axis.title.y = element_text(size = 24)) +
  theme_minimal() +
  labs(
    x = paste0("PC1 (", round(var_explained[1], 1), "%)"),
    y = paste0("PC2 (", round(var_explained[2], 1), "%)"),
    title = "PCA of Allele Frequencies"
  )

```
```{r}
png(filename="PC1.png", type="cairo",units="px", width=5400, height=3000, res=300, bg="transparent")
ggplot(pca_df, aes(x=PC1, y= PC2, color = Treatments, fill=Treatments)) + geom_point(aes(alpha=0.2), size =10,shape=21, col="black")+ theme_classic() + scale_fill_manual(values=cbPaletteSmall,name="Treatment")+ 
guides(alpha=FALSE) +theme(axis.title.x = element_text(size = 24),axis.title.y = element_text(size = 24))+
  labs(
    x = paste0("PC1 (", round(var_explained[1], 1), "%)"),
    y = paste0("PC2 (", round(var_explained[2], 1), "%)"),
    title = "PCA of Allele Frequencies"
  )
dev.off()
png(filename="PC1b.png", type="cairo",units="px", width=5400, height=3000, res=300, bg="transparent")
ggplot(pca_df, aes(x=PC1, y= PC2, color = Population, fill=Population)) + geom_point(aes(alpha=0.2), size =10,shape=21, col="black")+ theme_classic() + scale_fill_manual(values=cbPaletteSmall,name="Spawn")+ 
guides(alpha=FALSE) +theme(axis.title.x = element_text(size = 24),axis.title.y = element_text(size = 24))+
  labs(
    x = paste0("PC1 (", round(var_explained[1], 1), "%)"),
    y = paste0("PC2 (", round(var_explained[2], 1), "%)"),
    title = "PCA of Allele Frequencies"
  )
dev.off()
```
## Outlier loci for each individual blocks
```{bash, eval = FALSE}
source activate CASE


mawk 'NR==FNR{a[$1,$2]; next} ($1,$2) in a' TSL.pca.filter CASE.All.Blocks.sync | cut --complement -f60- > outlier.pca.block.sync
```


```{r}
b10.pos <- read.table("B10.pos")
b11.pos <- read.table("B11.pos")
b12.pos <- read.table("B12.pos")

input.all.outlier.block <- read.sync(file="outlier.pca.block.sync", gen=seq(1,56), repl=seq(1,56), polarization="rising")

treatment.labels <- c(rep("CA", 11),rep("CASE", 11),rep("CON", 11),rep("IS", 12),rep("SE", 11))
spawn.names <- c("B12","B11","B12","B10","B11","B10","B11","B12","B10","B12","B11","B10","B11","B12","B12","B10","B12","B11","B10","B11","B11","B12","B10","B11","B12","B11","B10","B12","B10","B11","B12","B11","B12","B10","B11","B10","B11","B10","B11","B10","B11","B12","B12","B12","B12","B11","B12","B10","B10","B11","B11","B10","B12","B12","B11","B12")
input.all.outlier.block.af <- af(input.all.outlier.block,gen=seq(1,56), repl=seq(1,56))
colnames(input.all.outlier.block.af) <- paste(treatment.labels,spawn.names,sep="_")

```


```{r}

colnames(input.all.outlier.block.af) <- trimws(colnames(input.all.outlier.block.af))

stages <- c("B10","B11","B12")

for(stage in stages){
  # Get the position data for this stage
  pos_data <- get(paste0(tolower(stage), ".pos"))
  
  # Use the chromosome identifiers directly from .pos file (no mapping needed)
  stage_loci <- paste(pos_data[,1], pos_data[,2], sep = ".")
  
  # Filter rows based on the constructed loci identifiers
  row_idx <- rownames(input.all.outlier.block.af) %in% stage_loci
  
  # Match columns that end with "_B10" / "_B11" / "_B12"
  cols_idx <- grep(paste0("_", stage, "$"), colnames(input.all.outlier.block.af))
  
  cat(sprintf("\n== Stage %s ==\n", stage))
  cat("  Found loci in .pos file:", length(stage_loci), "\n")
  cat("  Matching loci in data:", sum(row_idx), "\n")
  cat("  Found sample columns:", length(cols_idx), "\n")
  
  if(length(cols_idx) < 2){
    warning(sprintf("Stage %s: less than 2 samples found - skipping PCA.", stage))
    next
  }
  
  if(sum(row_idx) < 2){
    warning(sprintf("Stage %s: less than 2 matching loci found - skipping PCA.", stage))
    next
  }
  
  # Apply both row and column filtering
  sub_mat <- input.all.outlier.block.af[row_idx, cols_idx, drop = FALSE]
  cat("  dim(sub_mat) (loci x samples):", dim(sub_mat), "\n")
  
  # Transpose so rows = samples, cols = loci (variables)
  mat_t <- t(sub_mat)
  
  # Remove loci (columns) with zero variance across the samples in this stage
  var_cols <- apply(mat_t, 2, var, na.rm = TRUE)
  keep <- which(var_cols > 0)
  removed_n <- sum(var_cols == 0)
  cat("  loci removed (zero variance):", removed_n, "\n")
  
  if(length(keep) < 2){
    warning(sprintf("Stage %s: fewer than 2 variable loci after filtering - skipping PCA.", stage))
    next
  }
  
  mat_t2 <- mat_t[, keep, drop = FALSE]
  cat("  dim(mat_t2) (samples x retained loci):", dim(mat_t2), "\n")
  
  # Run PCA (now safe to scale)
  pca_res <- prcomp(mat_t2, center = TRUE, scale. = FALSE)
  
  # PCA dataframe
  pca_df <- as.data.frame(pca_res$x)
  
  # Extract Treatments and Spawn from sample names (rownames of mat_t2)
  sample_names <- rownames(mat_t2)
  pca_df$Treatment <- sub("_.*", "", sample_names)   # text before first "_"
  pca_df$Spawn     <- sub(".*_", "", sample_names)   # text after last "_"
  
  # Variance explained
  var_explained <- (pca_res$sdev^2) / sum(pca_res$sdev^2) * 100
  
  # Choose palette fallback if cbPaletteSmall not defined
  if(exists("cbPaletteSmall")){
    fill_scale <- scale_fill_manual(values = cbPaletteSmall, name = "Treatment")
  } else {
    fill_scale <- scale_fill_brewer(palette = "Set1", name = "Treatment")
  }
  
  # Plot
  p <- ggplot(pca_df, aes(x = PC1, y = PC2, fill = Treatment)) +
    geom_point(aes(alpha = 0.2), size = 6, shape = 21, color = "black") +
    fill_scale +
    guides(alpha = FALSE) +
    theme_classic() +
    theme(axis.title.x = element_text(size = 14),
          axis.title.y = element_text(size = 14),
          plot.title = element_text(hjust = 0.5)) +
    labs(
      title = paste0("PCA - ", stage),
      x = paste0("PC1 (", round(var_explained[1], 1), "%)"),
      y = paste0("PC2 (", round(var_explained[2], 1), "%)")
    )
  
  # Save (adjust size/resolution or use png(...) if you prefer)
  outfile <- paste0("PC_", stage, ".png")
  ggsave(outfile, plot = p, width = 12, height = 7, dpi = 300)
  cat("  saved:", outfile, "\n")
}
```

```{r}

# Base R approach for matrix named input.all.outlier.af
# Get column indices for each group
is_b10_cols <- grep("^IS_B10", colnames(input.all.outlier.af))
is_b11_cols <- grep("^IS_B11", colnames(input.all.outlier.af))
is_b12_cols <- grep("^IS_B12", colnames(input.all.outlier.af))

# Calculate row means and add as new columns
input.all.outlier.af <- cbind(input.all.outlier.af, 
                              ISM_B10 = rowMeans(input.all.outlier.af[, is_b10_cols], na.rm = TRUE),
                              ISM_B11 = rowMeans(input.all.outlier.af[, is_b11_cols], na.rm = TRUE),
                              ISM_B12 = rowMeans(input.all.outlier.af[, is_b12_cols], na.rm = TRUE))
```

```{r}
# Create new matrix with only delta columns (ISM values subtracted from corresponding B10/B11/B12 columns)
# (excluding IS columns from the subtraction)

# Get column indices for each group (excluding IS columns)
b10_cols <- grep("B10$", colnames(input.all.outlier.af))
b10_cols <- b10_cols[!grepl("^IS", colnames(input.all.outlier.af)[b10_cols])]

b11_cols <- grep("B11$", colnames(input.all.outlier.af))
b11_cols <- b11_cols[!grepl("^IS", colnames(input.all.outlier.af)[b11_cols])]

b12_cols <- grep("B12$", colnames(input.all.outlier.af))
b12_cols <- b12_cols[!grepl("^IS", colnames(input.all.outlier.af)[b12_cols])]

# Combine all target columns
all_target_cols <- c(b10_cols, b11_cols, b12_cols)

# Create vectors to store results and column names
delta_values <- list()
delta_names <- character()

# Process B10 columns
for(col in b10_cols) {
  delta_values[[length(delta_values) + 1]] <- input.all.outlier.af[, "ISM_B10"] - input.all.outlier.af[, col]
  delta_names <- c(delta_names, paste0(colnames(input.all.outlier.af)[col], "_delta"))
}

# Process B11 columns
for(col in b11_cols) {
  delta_values[[length(delta_values) + 1]] <- input.all.outlier.af[, "ISM_B11"] - input.all.outlier.af[, col]
  delta_names <- c(delta_names, paste0(colnames(input.all.outlier.af)[col], "_delta"))
}

# Process B12 columns
for(col in b12_cols) {
  delta_values[[length(delta_values) + 1]] <- input.all.outlier.af[, "ISM_B12"] - input.all.outlier.af[, col]
  delta_names <- c(delta_names, paste0(colnames(input.all.outlier.af)[col], "_delta"))
}

# Create the final matrix
new_matrix <- do.call(cbind, delta_values)
rownames(new_matrix) <- rownames(input.all.outlier.af)
colnames(new_matrix) <- delta_names

```



```{r}
# Perform PCA
af_t <- t(new_matrix)

pca_res <- prcomp(af_t, center = TRUE, scale. = FALSE)

# Get scores for plotting
pca_df <- as.data.frame(pca_res$x)
#pca_df$locus <- rownames(input.all.outlier.af)  # keep locus IDs if you have them

var_explained <- pca_res$sdev^2 / sum(pca_res$sdev^2) * 100

# Example: if you have a vector "pop_group" the same length as rows

treatment.labels <- c(rep("CA", 3),rep("CASE", 3),rep("CON", 3),rep("SE", 3),rep("CA", 4),rep("CASE", 4),rep("CON", 4),rep("SE", 4),rep("CA", 4),rep("CASE", 4),rep("CON", 4),rep("SE", 4))
spawn.names <- c(rep("B10",12),rep("B11",16),rep("B12",16))

pca_df$Population <- spawn.names
pca_df$Treatments <- treatment.labels

ggplot(pca_df, aes(x = PC1, y = PC2, fill = Treatments, shape = as.factor(Population))) +
  geom_point(size = 8, color = "black", alpha = 0.5) + 
  scale_fill_manual(values = cbPaletteSmall[-4], name = "Treatment") + 
  scale_shape_manual(values = c(21, 22, 23), name = "Spawn") +
  guides(fill = guide_legend(override.aes = list(shape = 21)), shape= guide_legend(override.aes= list(fill=cbPaletteSmall[4]))) +
  theme_minimal() +
  theme(axis.title.x = element_text(size = 24), axis.title.y = element_text(size = 24)) +
  labs(
    x = paste0("PC1 (", round(var_explained[1], 1), "%)"),
    y = paste0("PC2 (", round(var_explained[2], 1), "%)"),
    title = "PCA of Allele Frequencies"
  )

```
```{r}
library(ggplot2)

af_t <- t(new_matrix)  # rows = samples

meta <- data.frame(
  Spawn = spawn.names,
  Treatment = treatment.labels,
  stringsAsFactors = FALSE
)

stopifnot(nrow(af_t) == nrow(meta))

idx_by_spawn <- split(seq_len(nrow(af_t)), meta$Spawn)

run_pca_and_plot_one_spawn <- function(idx, spawn_id) {
  X <- af_t[idx, , drop = FALSE]
  meta_spawn <- meta[idx, , drop = FALSE]

  # Keep only numeric columns (in case something non-numeric slipped in)
  X <- X[, vapply(as.data.frame(X), is.numeric, logical(1)), drop = FALSE]

  # Drop columns that are all NA
  keep_not_all_na <- colSums(!is.na(X)) > 0
  X <- X[, keep_not_all_na, drop = FALSE]

  # Drop zero-variance columns (within this spawn)
  sds <- apply(X, 2, sd, na.rm = TRUE)
  keep_var <- is.finite(sds) & sds > 0
  X <- X[, keep_var, drop = FALSE]

  if (ncol(X) < 2) {
    stop(sprintf("Spawn %s: <2 variable columns remain after filtering; cannot run PCA.", spawn_id))
  }

  # PCA (now scaling is safe)
  pca_res <- prcomp(X, center = TRUE, scale. = FALSE)

  pca_df <- as.data.frame(pca_res$x)
  pca_df$Treatment <- meta_spawn$Treatment
  pca_df$Spawn <- spawn_id

  var_explained <- (pca_res$sdev^2 / sum(pca_res$sdev^2)) * 100

  ggplot(pca_df, aes(PC1, PC2, fill = Treatment)) +
    geom_point(size = 8, color = "black", alpha = 0.5, shape = 21) +
    scale_fill_manual(values = cbPaletteSmall[-4], name = "Treatment") +
    theme_minimal() +
    theme(axis.title.x = element_text(size = 24),
          axis.title.y = element_text(size = 24)) +
    labs(
      x = paste0("PC1 (", round(var_explained[1], 1), "%)"),
      y = paste0("PC2 (", round(var_explained[2], 1), "%)"),
      title = paste0("PCA of Allele Frequencies - ", spawn_id)
    )
}

plots_by_spawn <- mapply(
  FUN = run_pca_and_plot_one_spawn,
  idx = idx_by_spawn,
  spawn_id = names(idx_by_spawn),
  SIMPLIFY = FALSE
)

lapply(plots_by_spawn, print)

```




```{r}
library(ggplot2)

delta_mat <- new_matrix
spawns <- c("B10","B11","B12")

# mean across duplicate-named replicate columns
row_mean_by_exact_name <- function(mat, nm) {
  idx <- which(colnames(mat) == nm)
  if (length(idx) == 0) stop(paste("No columns found named:", nm))
  rowMeans(as.matrix(mat[, idx, drop = FALSE]), na.rm = TRUE)
}

make_pair_plot <- function(d, xvar, yvar, title_prefix) {
  r <- cor(d[[xvar]], d[[yvar]], use = "complete.obs", method = "pearson")
  n <- sum(complete.cases(d[[xvar]], d[[yvar]]))

  ggplot(d, aes(x = .data[[xvar]], y = .data[[yvar]])) +
    geom_hline(yintercept = 0, color = "grey75", linewidth = 0.4) +
    geom_vline(xintercept = 0, color = "grey75", linewidth = 0.4) +
    geom_point(alpha = 0.25, size = 0.9) +
    theme_minimal() +
    labs(
      title = paste0(title_prefix, " | r = ", round(r, 3), " | n = ", n),
      x = xvar,
      y = yvar
    )
}

make_color_plot <- function(d, spawn) {
  r <- cor(d$CA, d$SE, use = "complete.obs", method = "pearson")
  n <- sum(complete.cases(d$CA, d$SE, d$CASE))

  ggplot(d, aes(x = CA, y = SE, color = CASE)) +
    geom_hline(yintercept = 0, color = "grey75", linewidth = 0.4) +
    geom_vline(xintercept = 0, color = "grey75", linewidth = 0.4) +
    geom_point(alpha = 0.85, size = 1.2) +
    scale_color_gradient2(
      low = "#2b83ba", mid = "grey90", high = "#d7191c", midpoint = 0
    ) +
    theme_minimal() +
    labs(
      title = paste0("Spawn ", spawn, ": AF CA vs AF SE (color = AF CASE) | r(CA,SE) = ", round(r, 3), " | n = ", n),
      x = "AF CA",
      y = "AF SE",
      color = "AF CASE"
    )
}

plots <- list()

for (sp in spawns) {

  CA   <- row_mean_by_exact_name(delta_mat, paste0("CA_", sp, "_delta"))
  CASE <- row_mean_by_exact_name(delta_mat, paste0("CASE_", sp, "_delta"))
  SE   <- row_mean_by_exact_name(delta_mat, paste0("SE_", sp, "_delta"))

  d <- data.frame(
    SNP = rownames(delta_mat),
    CA = CA,
    CASE = CASE,
    SE = SE,
    stringsAsFactors = FALSE
  )
  d <- d[complete.cases(d$CA, d$CASE, d$SE), ]

  p_case_ca <- make_pair_plot(d, "CA", "CASE", paste0("Spawn ", sp, ": AF CASE vs AF CA"))
  p_case_se <- make_pair_plot(d, "SE", "CASE", paste0("Spawn ", sp, ": AF CASE vs AF SE"))
  p_ca_se_c <- make_color_plot(d, sp)

  plots[[sp]] <- list(
    CASE_vs_CA = p_case_ca,
    CASE_vs_SE = p_case_se,
    CA_vs_SE_col_CASE = p_ca_se_c
  )

  # Print the three plots for this spawn
  print(p_case_ca)
  print(p_case_se)
  print(p_ca_se_c)
}

# Access later, e.g.:
# plots[["B10"]]$CA_vs_SE_col_CASE
```

```{r}
library(ggplot2)

delta_mat <- new_matrix
colnames(delta_mat) <- trimws(colnames(delta_mat))
spawns <- c("B10","B11","B12")

# mean across duplicate-named replicate columns
row_mean_by_exact_name <- function(mat, nm) {
  idx <- which(colnames(mat) == nm)
  if (length(idx) == 0) stop(paste("No columns found named:", nm))
  rowMeans(as.matrix(mat[, idx, drop = FALSE]), na.rm = TRUE)
}

make_pair_plot <- function(d, xvar, yvar, title_prefix) {
  r <- cor(d[[xvar]], d[[yvar]], use = "complete.obs", method = "pearson")
  n <- sum(complete.cases(d[[xvar]], d[[yvar]]))

  ggplot(d, aes(x = .data[[xvar]], y = .data[[yvar]])) +
    geom_hline(yintercept = 0, color = "grey75", linewidth = 0.4) +
    geom_vline(xintercept = 0, color = "grey75", linewidth = 0.4) +
    geom_abline(slope = 1, intercept = 0, linetype = "dashed") +
    geom_point(alpha = 0.25, size = 0.9) +
    theme_minimal() +
    labs(
      title = paste0(title_prefix, " | r = ", round(r, 3), " | n = ", n),
      x = xvar,
      y = yvar
    )
}

make_color_plot <- function(d, spawn) {
  r <- cor(d$CA, d$SE, use = "complete.obs", method = "pearson")
  n <- sum(complete.cases(d$CA, d$SE, d$CASE))

  ggplot(d, aes(x = CA, y = SE, color = CASE)) +
    geom_hline(yintercept = 0, color = "grey75", linewidth = 0.4) +
    geom_vline(xintercept = 0, color = "grey75", linewidth = 0.4) +
    geom_point(alpha = 0.85, size = 1.2) +
    geom_abline(slope = 1, intercept = 0, linetype = "dashed") +
    scale_color_gradient2(low = "#2b83ba", mid = "grey90", high = "#d7191c", midpoint = 0) +
    theme_minimal() +
    labs(
      title = paste0(
        "Spawn ", spawn, ": AF CA vs AF SE (color = AF CASE) | r(CA,SE) = ",
        round(r, 3), " | n = ", n
      ),
      x = "AF CA",
      y = "AF SE",
      color = "AF CASE"
    )
}

plots <- list()

for (sp in spawns) {

  # --- FILTER ROWS BY SPAWN'S .pos LOCI (already loaded objects) ---
  pos_data <- get(paste0(tolower(sp), ".pos"))  # b10.pos, b11.pos, b12.pos
  stage_loci <- paste(pos_data[, 1], pos_data[, 2], sep = ".")

  sub_mat <- delta_mat[rownames(delta_mat) %in% stage_loci, , drop = FALSE]

  cat("\\n== Spawn", sp, "==\\n")
  cat("  loci in .pos:", length(stage_loci), "\\n")
  cat("  loci kept in matrix:", nrow(sub_mat), "\\n")

  # --- Compute replicate means on the filtered matrix ---
  CA   <- row_mean_by_exact_name(sub_mat, paste0("CA_", sp, "_delta"))
  CASE <- row_mean_by_exact_name(sub_mat, paste0("CASE_", sp, "_delta"))
  SE   <- row_mean_by_exact_name(sub_mat, paste0("SE_", sp, "_delta"))

  d <- data.frame(
    SNP = rownames(sub_mat),
    CA = CA,
    CASE = CASE,
    SE = SE,
    stringsAsFactors = FALSE
  )
  d <- d[complete.cases(d$CA, d$CASE, d$SE), ]

  p_case_ca <- make_pair_plot(d, "CA", "CASE", paste0("Spawn ", sp, ": AF CASE vs AF CA"))
  p_case_se <- make_pair_plot(d, "SE", "CASE", paste0("Spawn ", sp, ": AF CASE vs AF SE"))
  p_ca_se_c <- make_color_plot(d, sp)

  plots[[sp]] <- list(
    CASE_vs_CA = p_case_ca,
    CASE_vs_SE = p_case_se,
    CA_vs_SE_col_CASE = p_ca_se_c,
    data = d
  )

  print(p_case_ca)
  print(p_case_se)
  print(p_ca_se_c)
}

# Example:
# plots[["B10"]]$CA_vs_SE_col_CASE

```
```{r}
library(ggplot2)

af_mat <- input.all.outlier.af   # or input.all.rand.af
spawns <- c("B10", "B11", "B12")

# Mean across ALL replicate columns with the exact same name (handles duplicate colnames)
af_mean <- function(mat, trt, sp) {
  nm <- paste0(trt, "_", sp)
  idx <- which(colnames(mat) == nm)
  if (length(idx) == 0) stop(paste("No columns found named:", nm))
  rowMeans(as.matrix(mat[, idx, drop = FALSE]), na.rm = TRUE)
}

pair_plot <- function(d, x, y, title) {
  r <- cor(d[[x]], d[[y]], use = "complete.obs", method = "pearson")
  n <- sum(complete.cases(d[[x]], d[[y]]))

  ggplot(d, aes(x = .data[[x]], y = .data[[y]])) +
    geom_point(alpha = 0.30, size = 1.0) +
    theme_minimal() +
    labs(
      title = paste0(title, " | r = ", round(r, 3), " | n = ", n),
      x = paste0("AF ", x, " (mean across reps)"),
      y = paste0("AF ", y, " (mean across reps)")
    )
}

color_plot <- function(d, sp) {
  r <- cor(d$CA, d$SE, use = "complete.obs", method = "pearson")
  n <- sum(complete.cases(d$CA, d$SE, d$CASE))

  ggplot(d, aes(x = CA, y = SE, color = CASE)) +
    geom_point(alpha = 0.55, size = 1.2) +
    scale_color_gradient2(
      low = "#2b83ba", mid = "grey90", high = "#d7191c",
      midpoint = median(d$CASE, na.rm = TRUE)
    ) +
    theme_minimal() +
    labs(
      title = paste0("Spawn ", sp, ": CA vs SE (color = CASE) | r(CA,SE) = ", round(r, 3), " | n = ", n),
      x = "AF CA (mean across reps)",
      y = "AF SE (mean across reps)",
      color = "AF CASE\\n(mean)"
    )
}

plots <- list()

for (sp in spawns) {

  # Average across replicates within each treatmentspawn
  d <- data.frame(
    SNP  = rownames(af_mat),
    CA   = af_mean(af_mat, "CA", sp),
    CASE = af_mean(af_mat, "CASE", sp),
    SE   = af_mean(af_mat, "SE", sp),
    stringsAsFactors = FALSE
  )
  d <- d[complete.cases(d$CA, d$CASE, d$SE), ]

  p_case_ca <- pair_plot(d, x = "CA", y = "CASE", title = paste0("Spawn ", sp, ": CASE vs CA (AF)"))
  p_case_se <- pair_plot(d, x = "SE", y = "CASE", title = paste0("Spawn ", sp, ": CASE vs SE (AF)"))
  p_ca_se_c <- color_plot(d, sp)

  plots[[sp]] <- list(
    CASE_vs_CA = p_case_ca,
    CASE_vs_SE = p_case_se,
    CA_vs_SE_col_CASE = p_ca_se_c,
    data = d
  )

  print(p_case_ca)
  print(p_case_se)
  print(p_ca_se_c)
}

# Example access:
# plots[["B10"]]$data
# plots[["B12"]]$CA_vs_SE_col_CASE
```


```{r}
library(ggplot2)

colnames(input.all.outlier.af) <- trimws(colnames(input.all.outlier.af))

stages <- c("B10","B11","B12")

subset_by_stage <- function(mat, stage) {
  pos_data <- get(paste0(tolower(stage), ".pos"))   # b10.pos, b11.pos, b12.pos
  stage_loci <- paste(pos_data[, 1], pos_data[, 2], sep = ".")

  row_idx  <- rownames(mat) %in% stage_loci
  cols_idx <- grep(paste0(stage, "$"), colnames(mat))  # columns ending with B10/B11/B12

  cat(sprintf("\\n== Stage %s ==\\n", stage))
  cat("  Found loci in .pos file:", length(stage_loci), "\\n")
  cat("  Matching loci in data:", sum(row_idx), "\\n")
  cat("  Found sample columns:", length(cols_idx), "\\n")

  mat[row_idx, cols_idx, drop = FALSE]
}

mean_by_exact_colname <- function(mat, colname_exact) {
  idx <- which(colnames(mat) == colname_exact)  # handles duplicate column names
  if (length(idx) == 0) stop(paste("No columns found named:", colname_exact))
  rowMeans(as.matrix(mat[, idx, drop = FALSE]), na.rm = TRUE)
}

plots <- list()

for (stage in stages) {

  sub_mat <- subset_by_stage(input.all.outlier.af, stage)

  # mean AF across replicates for each treatment at this stage
  d <- data.frame(
    SNP  = rownames(sub_mat),
    CA   = mean_by_exact_colname(sub_mat, paste0("CA_", stage)),
    CASE = mean_by_exact_colname(sub_mat, paste0("CASE_", stage)),
    SE   = mean_by_exact_colname(sub_mat, paste0("SE_", stage)),
    stringsAsFactors = FALSE
  )
  d <- d[complete.cases(d$CA, d$CASE, d$SE), ]

  r_case_ca <- cor(d$CASE, d$CA, use = "complete.obs")
  r_case_se <- cor(d$CASE, d$SE, use = "complete.obs")
  r_ca_se   <- cor(d$CA, d$SE,   use = "complete.obs")

  p1 <- ggplot(d, aes(x = CA, y = CASE)) +
    geom_point(alpha = 0.30, size = 1.0) +
    geom_abline(slope = 1, intercept = 0, linetype = "dashed") +
    theme_minimal() +
    labs(
      title = paste0(stage, " AF (filtered loci): CASE vs CA | r=", round(r_case_ca, 3)),
      x = "CA AF (mean across reps)",
      y = "CASE AF (mean across reps)"
    )

  p2 <- ggplot(d, aes(x = SE, y = CASE)) +
    geom_point(alpha = 0.30, size = 1.0) +
    geom_abline(slope = 1, intercept = 0, linetype = "dashed")+
    theme_minimal() +
    labs(
      title = paste0(stage, " AF (filtered loci): CASE vs SE | r=", round(r_case_se, 3)),
      x = "SE AF (mean across reps)",
      y = "CASE AF (mean across reps)"
    )

  p3 <- ggplot(d, aes(x = CA, y = SE, color = CASE)) +
    geom_point(alpha = 0.55, size = 1.1) +
    geom_abline(slope = 1, intercept = 0, linetype = "dashed")+
    scale_color_gradient2(
      low = "#2b83ba", mid = "grey90", high = "#d7191c",
      midpoint = median(d$CASE, na.rm = TRUE)
    ) +
    theme_minimal() +
    labs(
      title = paste0(stage, " AF (filtered loci): CA vs SE (color=CASE) | r(CA,SE)=", round(r_ca_se, 3)),
      x = "CA AF (mean across reps)",
      y = "SE AF (mean across reps)",
      color = "CASE AF\\n(mean)"
    )

  print(p1); print(p2); print(p3)

  plots[[stage]] <- list(data = d, CASE_vs_CA = p1, CASE_vs_SE = p2, CA_vs_SE_col_CASE = p3)
}
```


